<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>Phylogenetics | Recology</title>
<meta name="keywords" content="">
<meta name="description" content="">
<meta name="author" content="">
<link rel="canonical" href="http://localhost:1313/tags/phylogenetics/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.f4cda22486791d430ef692108ba33a982c3c552eca57e99456bb55aa4bd22568.css" integrity="sha256-9M2iJIZ5HUMO9pIQi6M6mCw8VS7KV&#43;mUVrtVqkvSJWg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" type="application/rss+xml" href="http://localhost:1313/tags/phylogenetics/index.xml">
<link rel="alternate" hreflang="en" href="http://localhost:1313/tags/phylogenetics/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="list" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Recology (Alt + H)">Recology</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/" title="Home">
                    <span>Home</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/index.xml" title="RSS">
                    <span>RSS</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main"> 
<header class="page-header">
  <h1>
    Phylogenetics
  </h1>
</header>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Running Phylip&#39;s contrast application for trait pairs from R
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is some code to run Phylip’s contrast application from R and get the output within R to easily manipulate yourself. Importantly, the code is written specifically for trait pairs only as the regular expression work in the code specifically grabs data from contast results when only two traits are input. You could easily change the code to do N traits. Note that the p-value calculated for the chi-square statistic is not output from contrast, but is calculated within the function ‘PhylipWithinSpContr’....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-26 07:40:00.004 -0500 -0500'>April 26, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Running Phylip&#39;s contrast application for trait pairs from R" href="http://localhost:1313/2011/04/running-phylip-s-contrast-application-for-trait-pairs-from-r/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R: Randomization via Tip Shuffle
    </h2>
  </header>
  <div class="entry-content">
    <p>—UPDATE: I am now using code formatting from gist.github, so I replaced the old prettyR code (sorry guys). The github way is much easier and prettier. I hope readers like the change.
I wrote earlier about some code I wrote for running Phylometa (software to do phylogenetic meta-analysis) from R.
I have been concerned about what exactly is the right penalty for including phylogeny in a meta-analysis. E.g.: AIC is calculated from Q in Phylometa, and Q increases with tree size....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-16 13:44:00.004 -0500 -0500'>April 16, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R: Randomization via Tip Shuffle" href="http://localhost:1313/2011/04/phylometa-from-r-randomization-via-tip-shuffle/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Adjust branch lengths with node ages: comparison of two methods
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is an approach for comparing two methods of adjusting branch lengths on trees: bladj in the program Phylocom and a fxn written by Gene Hunt at the Smithsonian.
Get the code and example files (tree and node ages) at https://gist.github.com/938313
Get phylocom at http://www.phylodiversity.net/phylocom/
Gene Hunt’s method has many options you can mess with, including setting tip ages (not available in bladj), setting node ages, and minimum branch length imposed....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-10 13:02:00.003 -0500 -0500'>April 10, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Adjust branch lengths with node ages: comparison of two methods" href="http://localhost:1313/2011/04/adjust-branch-lengths-with-node-ages-comparison-of-two-methods/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R - UDPATE
    </h2>
  </header>
  <div class="entry-content">
    <p>A while back I posted some messy code to run Phylometa from R, especially useful for processing the output data from Phylometa which is not easily done. The code is still quite messy, but it should work now. I have run the code with tens of different data sets and phylogenies so it should work.
I fixed errors when parentheses came up against numbers in the output, and other things. You can use the code for up to 4 levels of your grouping variable....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-01 16:18:00.003 -0500 -0500'>April 1, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R - UDPATE" href="http://localhost:1313/2011/04/phylometa-from-r-udpate/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Check out Phyloseminar.org
    </h2>
  </header>
  <div class="entry-content">
    <p>They have online seminars that you can join in on live, and watch later as recorded videos. Check it out at phyloseminar.org</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-04 08:22:00 -0600 -0600'>March 4, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Check out Phyloseminar.org" href="http://localhost:1313/2011/03/check-out-phyloseminar-org/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylogenetic analysis with the phangorn package: an example
    </h2>
  </header>
  <div class="entry-content">
    <p>The phangorn package is a relatively new package in R for the analysis and comparison of phylogenies. See here for the Bioinformatics paper and here for the package. Here is an example of using phangorn from getting sequences to making phylogenies and visualizing them:Getting sequences from GenbankMultiple alignmentMaximum likelihood tree reconstructionVisualizing treesVisualizing trees and traitsMake fake traits:Visualize them on trees:</p>
  </div>
  <footer class="entry-footer"><span title='2011-02-21 12:31:00.001 -0600 -0600'>February 21, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylogenetic analysis with the phangorn package: an example" href="http://localhost:1313/2011/02/phylogenetic-analysis-with-the-phangorn-package-an-example/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">New approach to analysis of phylogenetic community structure
    </h2>
  </header>
  <div class="entry-content">
    <p>Anthony Ives, of University of Wisconsin-Madison, and Matthew Helmus of the Xishuangbanna Tropical Botanical Garden, present a new statistical method for analyzing phylogenetic community structure in an early view paper in Ecological Monographs. See the abstract here.
Up to now, most phylogenetic community structure papers have calculated metrics and used randomization tests to determine if observed metrics are different from random. The approach of Ives and Helmus fits models to observed data, instead of calculating single metrics....</p>
  </div>
  <footer class="entry-footer"><span title='2011-01-05 07:54:00.001 -0600 -0600'>January 5, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to New approach to analysis of phylogenetic community structure" href="http://localhost:1313/2011/01/new-approach-to-analysis-of-phylogenetic-community-structure/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylogenetic meta-analysis in R using Phylometa
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is some code to run Phylometa from R. Phylometa is a program that conducts phylogenetic meta-analyses. The great advantage of the approach below is that you can easily run Phylometa from R, and manipulate the output from Phylometa in R.
Phylometa was created by Marc Lajeunesse at University of South Florida, and is described in his 2009 AmNat paper. Phylometa can be downloaded free here.
Save phylometa_fxn.R (get here) to your working directory....</p>
  </div>
  <footer class="entry-footer"><span title='2010-12-28 07:15:00.002 -0600 -0600'>December 28, 2010</span>&nbsp;·&nbsp;3 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylogenetic meta-analysis in R using Phylometa" href="http://localhost:1313/2010/12/phylogenetic-meta-analysis-in-r-using-phylometa/"></a>
</article>
<footer class="page-footer">
  <nav class="pagination">
    <a class="prev" href="http://localhost:1313/tags/phylogenetics/">
      «&nbsp;Prev&nbsp;
    </a>
  </nav>
</footer>
    </main>
    
<footer class="footer">
    <span>&copy; 2024 <a href="http://localhost:1313/">Recology</a></span>
    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
