<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
  <link href="http://gmpg.org/xfn/11" rel="profile">

  <title>
    
    Recology, R ecology
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:300,400italic,400,600,700|Abril+Fatface">


  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/favicon.ico">
  <link rel="shortcut icon" href="/public/favicon.ico">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.css" rel="stylesheet">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body class="theme-base-0f layout-reverse">

    <header class="masthead">
      <div class="masthead-inner">
        <h1>Recology</h1>
        <!-- <h1> <a href="http://recology.info/">Recology</a></h1> -->
        <p class="lead">R/etc.</p>

        <div class="colophon">
          <ul class="colophon-links">
            <li>
              <a href="/"><i class="fa fa-home fa-lg"></i></a>&nbsp;
              <a href="/about"><i class="fa fa-info-circle fa-lg"></i></a>&nbsp;
              <a href="/archives"><i class="fa fa-archive fa-lg"></i></a>&nbsp;
              <a href="/rresources"><i class="fa fa-book fa-lg"></i></a>&nbsp;
              <a href="http://rforcats.net/" rel><i class="fa fa-graduation-cap fa-lg"></i></a>&nbsp;
              <a href="http://feeds.feedburner.com/github/wpna"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
              <a href="https://twitter.com/recology_"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
              <a href="/fork"><i class="fa fa-spinner fa-lg"></i></a>
            </li>
          </ul>
          <!-- <small><a href="https://github.com/mdo/hyde">Hyde</a> from <a href="https://twitter.com/mdo" target="_blank">@mdo</a>.</small> -->
        </div>
      </div>
    </header>

    <div class="content container">
      <div class="posts">
  <a style="float:right;" href="/archives"><i class="fa fa-archive fa-lg"></i></a>
  
  <div class="post">
    <h1>
      <a href="/2012/08/ggplot-inset-map/">
        ggplot2 maps with insets
      </a>
    </h1>

    <span class="post-date">22 Aug 2012</span>

    <hr />

<h2>UPDATE: changed data source so that the entire example can be run by anone on their own machine. Also, per Joachim's suggestion, I put a box around the blown up area of the map. Thoughts?</h2>

<hr />

<h2>Here's a quick demo of creating a map with an inset within it using ggplot. The inset is achieved using the <code>gridExtra</code> package.</h2>

<hr />

<h3>Install libraries</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> <span class="kn">library</span><span class="p">(</span>ggplot2<span class="p">)</span>
<span class="lineno">2</span> <span class="kn">library</span><span class="p">(</span>maps<span class="p">)</span>
<span class="lineno">3</span> <span class="kn">library</span><span class="p">(</span>maptools<span class="p">)</span>
<span class="lineno">4</span> <span class="kn">library</span><span class="p">(</span>gridExtra<span class="p">)</span>
<span class="lineno">5</span> <span class="kn">library</span><span class="p">(</span>rgeos<span class="p">)</span></code></pre></div>


<hr />

<h3>Create a data frame</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> dat <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>ecosystem <span class="o">=</span> <span class="kp">rep</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;oak&quot;</span><span class="p">,</span> <span class="s">&quot;steppe&quot;</span><span class="p">,</span> <span class="s">&quot;prairie&quot;</span><span class="p">),</span> each <span class="o">=</span> <span class="m">8</span><span class="p">),</span> 
<span class="lineno">2</span>     lat <span class="o">=</span> rnorm<span class="p">(</span><span class="m">24</span><span class="p">,</span> mean <span class="o">=</span> <span class="m">51</span><span class="p">,</span> sd <span class="o">=</span> <span class="m">1</span><span class="p">),</span> lon <span class="o">=</span> rnorm<span class="p">(</span><span class="m">24</span><span class="p">,</span> mean <span class="o">=</span> <span class="m">-113</span><span class="p">,</span> sd <span class="o">=</span> <span class="m">5</span><span class="p">))</span>
<span class="lineno">3</span> <span class="kp">head</span><span class="p">(</span>dat<span class="p">)</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">ecosystem   lat    lon
1       oak 51.81 -109.1
2       oak 50.41 -115.0
3       oak 52.20 -112.5
4       oak 51.67 -114.8
5       oak 49.75 -105.4
6       oak 51.42 -114.6</code></pre></div>


<hr />

<h3>Get maps using the maps library</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno"> 1</span> <span class="c1"># Get a map of Canada</span>
<span class="lineno"> 2</span> canadamap <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>map<span class="p">(</span><span class="s">&quot;world&quot;</span><span class="p">,</span> <span class="s">&quot;Canada&quot;</span><span class="p">,</span> plot <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)[</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;x&quot;</span><span class="p">,</span> <span class="s">&quot;y&quot;</span><span class="p">)])</span>
<span class="lineno"> 3</span> 
<span class="lineno"> 4</span> <span class="c1"># Get a map of smaller extent</span>
<span class="lineno"> 5</span> canadamapsmall <span class="o">&lt;-</span> canadamap<span class="p">[</span>canadamap<span class="o">$</span>x <span class="o">&lt;</span> <span class="m">-90</span> <span class="o">&amp;</span> canadamap<span class="o">$</span>y <span class="o">&lt;</span> <span class="m">54</span><span class="p">,</span> <span class="p">]</span>
<span class="lineno"> 6</span> 
<span class="lineno"> 7</span> <span class="c1"># Make inset rectangle to show area of zoom</span>
<span class="lineno"> 8</span> canadamapsmall_ <span class="o">&lt;-</span> na.omit<span class="p">(</span>canadamapsmall<span class="p">)</span>  <span class="c1"># omit NA&#39;s</span>
<span class="lineno"> 9</span> 
<span class="lineno">10</span> <span class="c1"># This should get your corner points for the box, picking min and max of</span>
<span class="lineno">11</span> <span class="c1"># lat and lon</span>
<span class="lineno">12</span> insetrect <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>xmin <span class="o">=</span> <span class="kp">min</span><span class="p">(</span>canadamapsmall_<span class="o">$</span>x<span class="p">),</span> xmax <span class="o">=</span> <span class="kp">max</span><span class="p">(</span>canadamapsmall_<span class="o">$</span>x<span class="p">),</span> 
<span class="lineno">13</span>     ymin <span class="o">=</span> <span class="kp">min</span><span class="p">(</span>canadamapsmall_<span class="o">$</span>y<span class="p">),</span> ymax <span class="o">=</span> <span class="kp">max</span><span class="p">(</span>canadamapsmall_<span class="o">$</span>y<span class="p">))</span>
<span class="lineno">14</span> insetrect</code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">xmin   xmax  ymin ymax
1 -133.1 -90.39 48.05   54</code></pre></div>


<hr />

<h3>Make the maps</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno"> 1</span> <span class="c1"># The inset map, all of Canada</span>
<span class="lineno"> 2</span> a <span class="o">&lt;-</span> ggplot<span class="p">(</span>canadamap<span class="p">)</span> <span class="o">+</span> 
<span class="lineno"> 3</span>  theme_bw<span class="p">(</span>base_size <span class="o">=</span> <span class="m">22</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno"> 4</span>  geom_path<span class="p">(</span>data <span class="o">=</span> canadamap<span class="p">,</span> aes<span class="p">(</span>x<span class="p">,</span> y<span class="p">),</span> colour <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">,</span> fill <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno"> 5</span>  geom_rect<span class="p">(</span>data <span class="o">=</span> insetrect<span class="p">,</span> aes<span class="p">(</span>xmin <span class="o">=</span> xmin<span class="p">,</span> xmax <span class="o">=</span> xmax<span class="p">,</span> ymin <span class="o">=</span> ymin<span class="p">,</span> ymax <span class="o">=</span> ymax<span class="p">),</span> alpha<span class="o">=</span><span class="m">0</span><span class="p">,</span> colour<span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">,</span> size <span class="o">=</span> <span class="m">1</span><span class="p">,</span> linetype<span class="o">=</span><span class="m">1</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno"> 6</span>  opts<span class="p">(</span>panel.border <span class="o">=</span> theme_rect<span class="p">(</span>colour <span class="o">=</span> <span class="s">&#39;black&#39;</span><span class="p">,</span> size <span class="o">=</span> <span class="m">1</span><span class="p">,</span> linetype<span class="o">=</span><span class="m">1</span><span class="p">),</span>
<span class="lineno"> 7</span>           panel.grid.major <span class="o">=</span> theme_blank<span class="p">(),</span> panel.grid.minor<span class="o">=</span>theme_blank<span class="p">(),</span>
<span class="lineno"> 8</span>           panel.background <span class="o">=</span> theme_rect<span class="p">(</span> fill <span class="o">=</span> <span class="s">&#39;white&#39;</span><span class="p">),</span>
<span class="lineno"> 9</span>           legend.position <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">0.15</span><span class="p">,</span><span class="m">0.80</span><span class="p">),</span> legend.key <span class="o">=</span> theme_blank<span class="p">(),</span>
<span class="lineno">10</span>           axis.ticks <span class="o">=</span> theme_blank<span class="p">(),</span> axis.text.x<span class="o">=</span>theme_blank<span class="p">(),</span>
<span class="lineno">11</span>           axis.text.y<span class="o">=</span>theme_blank<span class="p">())</span> <span class="o">+</span>
<span class="lineno">12</span>  labs<span class="p">(</span>x <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> y <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
<span class="lineno">13</span> 
<span class="lineno">14</span> <span class="c1"># The larger map, zoomed in, with the data</span>
<span class="lineno">15</span> b <span class="o">&lt;-</span> ggplot<span class="p">(</span>dat<span class="p">,</span> aes<span class="p">(</span>lon<span class="p">,</span> lat<span class="p">,</span> colour<span class="o">=</span>ecosystem<span class="p">))</span> <span class="o">+</span>
<span class="lineno">16</span>  theme_bw<span class="p">(</span>base_size <span class="o">=</span> <span class="m">22</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno">17</span>  geom_jitter<span class="p">(</span>size<span class="o">=</span><span class="m">4</span><span class="p">,</span> alpha<span class="o">=</span><span class="m">0.6</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno">18</span>  geom_path<span class="p">(</span>data <span class="o">=</span> canadamapsmall<span class="p">,</span> aes<span class="p">(</span>x<span class="p">,</span> y<span class="p">),</span> colour <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">,</span> fill <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno">19</span>  scale_size<span class="p">(</span>guide<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
<span class="lineno">20</span>  opts<span class="p">(</span>panel.border <span class="o">=</span> theme_rect<span class="p">(</span>colour <span class="o">=</span> <span class="s">&#39;black&#39;</span><span class="p">,</span> size <span class="o">=</span> <span class="m">1</span><span class="p">,</span> linetype<span class="o">=</span><span class="m">1</span><span class="p">),</span>
<span class="lineno">21</span>           panel.grid.major <span class="o">=</span> theme_blank<span class="p">(),</span> panel.grid.minor<span class="o">=</span>theme_blank<span class="p">(),</span>
<span class="lineno">22</span>           panel.background <span class="o">=</span> theme_rect<span class="p">(</span> fill <span class="o">=</span> <span class="s">&#39;white&#39;</span><span class="p">),</span>
<span class="lineno">23</span>           legend.position <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="m">0.20</span><span class="p">),</span> legend.text<span class="o">=</span>theme_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span> face<span class="o">=</span><span class="s">&#39;bold&#39;</span><span class="p">),</span> 
<span class="lineno">24</span>           legend.title<span class="o">=</span>theme_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span> face<span class="o">=</span><span class="s">&#39;bold&#39;</span><span class="p">),</span> legend.key <span class="o">=</span> theme_blank<span class="p">(),</span>
<span class="lineno">25</span>           axis.ticks <span class="o">=</span> theme_segment<span class="p">(</span>size <span class="o">=</span> <span class="m">2</span><span class="p">))</span> <span class="o">+</span>
<span class="lineno">26</span>  labs<span class="p">(</span>x <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> y <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span></code></pre></div>


<hr />

<h3>Print the two maps together, one an inset of the other</h3>

<h4>This approach uses the <code>gridExtra</code> package for flexible alignment, etc. of ggplot graphs</h4>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> grid.newpage<span class="p">()</span>
<span class="lineno">2</span> vpb_ <span class="o">&lt;-</span> viewport<span class="p">(</span>width <span class="o">=</span> <span class="m">1</span><span class="p">,</span> height <span class="o">=</span> <span class="m">1</span><span class="p">,</span> x <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> y <span class="o">=</span> <span class="m">0.5</span><span class="p">)</span>  <span class="c1"># the larger map</span>
<span class="lineno">3</span> vpa_ <span class="o">&lt;-</span> viewport<span class="p">(</span>width <span class="o">=</span> <span class="m">0.4</span><span class="p">,</span> height <span class="o">=</span> <span class="m">0.4</span><span class="p">,</span> x <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> y <span class="o">=</span> <span class="m">0.8</span><span class="p">)</span>  <span class="c1"># the inset in upper right</span>
<span class="lineno">4</span> <span class="kp">print</span><span class="p">(</span>b<span class="p">,</span> vp <span class="o">=</span> vpb_<span class="p">)</span>
<span class="lineno">5</span> <span class="kp">print</span><span class="p">(</span>a<span class="p">,</span> vp <span class="o">=</span> vpa_<span class="p">)</span></code></pre></div>


<p><img src="/public/img/unnamed-chunk-5.png" alt="center" /></p>

<hr />

<h3>Get the .Rmd file used to create this post <a href="https://github.com/sckott/sckott.github.io/tree/master/_drafts/2012-08-22-ggplot-inset-map.Rmd">at my github account</a>.</h3>

<hr />

<h3>Written in <a href="http://daringfireball.net/projects/markdown/">Markdown</a>, with help from <a href="http://yihui.name/knitr/">knitr</a>, and nice knitr highlighting/etc. in in <a href="http://rstudio.org/">RStudio</a>.</h3>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/07/global-names-resolver/">
        Hitting the Global Names Resolver API
      </a>
    </h1>

    <span class="post-date">20 Jul 2012</span>

    <h2>Example of using the Global Names Resolver API to check species names</h2>

<hr />

<p>There are a number of options for resolution of taxonomic names. The <a href="http://tnrs.iplantcollaborative.org/">Taxonomic Name Resolution Service (TNRS)</a> comes to mind. There is a new service for taxonomic name resoultion called the <a href="http://resolver.globalnames.org/">Global Names Resolver</a>. They describe the service thusly "<em>Resolve lists of scientific names against known sources. This service parses incoming names, executes exact or fuzzy matching as required, and displays a confidence score for each match along with its identifier.</em>".</p>

<hr />

<h2>Load required packages</h2>

<h3>Just uncomment the code to use.</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> <span class="c1"># If you don&#39;t have them already</span>
<span class="lineno">2</span> <span class="c1"># install.packages(c(&#39;RJSONIO&#39;,&#39;plyr&#39;,&#39;devtools&#39;)) require(devtools)</span>
<span class="lineno">3</span> <span class="c1"># install_github(&#39;taxize_&#39;,&#39;ropensci&#39;)</span>
<span class="lineno">4</span> <span class="kn">library</span><span class="p">(</span>RJSONIO<span class="p">)</span>
<span class="lineno">5</span> <span class="kn">library</span><span class="p">(</span>plyr<span class="p">)</span>
<span class="lineno">6</span> <span class="kn">library</span><span class="p">(</span>taxize<span class="p">)</span></code></pre></div>


<h2>Get the data sources available</h2>

<h3>Get just id's and names of sources in a data.frame</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> <span class="kp">tail</span><span class="p">(</span>gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">))</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">##     id                                title
## 82 164                            BioLib.cz
## 83 165 Tropicos - Missouri Botanical Garden
## 84 166                                nlbif
## 85 167                                 IPNI
## 86 168              Index to Organism Names
## 87 169                        uBio NameBank</code></pre></div>


<hr />

<h3>Give me the id for EOL (Encyclopedia of Life)</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> out <span class="o">&lt;-</span> gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="lineno">2</span> out<span class="p">[</span>out<span class="o">$</span>title <span class="o">==</span> <span class="s">&quot;EOL&quot;</span><span class="p">,</span> <span class="s">&quot;id&quot;</span><span class="p">]</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">## [1] 12</code></pre></div>


<hr />

<h3>Fuzzy search for sources with the word "zoo"</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> out <span class="o">&lt;-</span> gnr_datasources<span class="p">(</span>todf <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="lineno">2</span> outdf <span class="o">&lt;-</span> out<span class="p">[</span><span class="kp">agrep</span><span class="p">(</span><span class="s">&quot;zoo&quot;</span><span class="p">,</span> out<span class="o">$</span>title<span class="p">,</span> ignore.case <span class="o">=</span> <span class="bp">T</span><span class="p">),</span> <span class="p">]</span>
<span class="lineno">3</span> outdf<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span> <span class="p">]</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">##     id             title
## 20 100 Mushroom Observer
## 25 105           ZooKeys</code></pre></div>


<h2>Resolve some names</h2>

<h3>Search for <em>Helianthus annuus</em> and <em>Homo sapiens</em>, return a data.frame</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> gnr<span class="p">(</span>names <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Helianthus annuus&quot;</span><span class="p">,</span> <span class="s">&quot;Homo sapiens&quot;</span><span class="p">),</span> returndf <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span> <span class="p">]</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">##   data_source_id    submitted_name       name_string score    title
## 1              4 Helianthus annuus Helianthus annuus 0.988     NCBI
## 3             10 Helianthus annuus Helianthus annuus 0.988 Freebase</code></pre></div>


<hr />

<h3>Search for the same species, with only using data source 12 (i.e., EOL)</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno">1</span> gnr<span class="p">(</span>names <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Helianthus annuus&quot;</span><span class="p">,</span> <span class="s">&quot;Homo sapiens&quot;</span><span class="p">),</span> data_source_ids <span class="o">=</span> <span class="s">&quot;12&quot;</span><span class="p">,</span> 
<span class="lineno">2</span>     returndf <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span></code></pre></div>




<div class="highlight"><pre><code class="language-text" data-lang="text">##   data_source_id    submitted_name       name_string score title
## 1             12 Helianthus annuus Helianthus annuus 0.988   EOL
## 2             12      Homo sapiens      Homo sapiens 0.988   EOL</code></pre></div>


<h3>That's it. Have fun! And put bugs/comments/etc. <a href="https://github.com/ropensci/taxize_/issues">here</a>.</h3>

<hr />

<h3>Written in <a href="http://daringfireball.net/projects/markdown/">Markdown</a>, with help from <a href="http://yihui.name/knitr/">knitr</a>, and nice knitr highlighting/etc. in in <a href="http://rstudio.org/">RStudio</a>.</h3>

<hr />

<h3>I prepared the markdown for this post by:</h3>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="lineno"> 1</span> KnitPost <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>input<span class="p">,</span> base.url <span class="o">=</span> <span class="s">&quot;/&quot;</span><span class="p">)</span> <span class="p">{</span>
<span class="lineno"> 2</span>     <span class="kn">require</span><span class="p">(</span>knitr<span class="p">)</span>
<span class="lineno"> 3</span>     opts_knit<span class="o">$</span>set<span class="p">(</span>base.url <span class="o">=</span> base.url<span class="p">)</span>
<span class="lineno"> 4</span>     fig.path <span class="o">&lt;-</span> <span class="kp">paste0</span><span class="p">(</span><span class="s">&quot;img/&quot;</span><span class="p">,</span> <span class="kp">sub</span><span class="p">(</span><span class="s">&quot;.Rmd$&quot;</span><span class="p">,</span> <span class="s">&quot;&quot;</span><span class="p">,</span> <span class="kp">basename</span><span class="p">(</span>input<span class="p">)),</span> <span class="s">&quot;/&quot;</span><span class="p">)</span>
<span class="lineno"> 5</span>     opts_chunk<span class="o">$</span>set<span class="p">(</span>fig.path <span class="o">=</span> fig.path<span class="p">)</span>
<span class="lineno"> 6</span>     opts_chunk<span class="o">$</span>set<span class="p">(</span>fig.cap <span class="o">=</span> <span class="s">&quot;center&quot;</span><span class="p">)</span>
<span class="lineno"> 7</span>     render_jekyll<span class="p">()</span>
<span class="lineno"> 8</span>     knit<span class="p">(</span>input<span class="p">,</span> envir <span class="o">=</span> <span class="kp">parent.frame</span><span class="p">())</span>
<span class="lineno"> 9</span> <span class="p">}</span>
<span class="lineno">10</span> <span class="kp">setwd</span><span class="p">(</span><span class="s">&quot;/path/to/_posts&quot;</span><span class="p">)</span>
<span class="lineno">11</span> KnitPost<span class="p">(</span><span class="s">&quot;/path/to/postfile.Rmd&quot;</span><span class="p">)</span></code></pre></div>


<p>from <a href="http://jfisher-usgs.github.com/r/2012/07/03/knitr-jekyll/">jfisher</a>.</p>

  </div>
  
  <div class="post">
    <h1>
      <a href="/2012/06/recent-r-eeb-packages/">
        Recent R packages for ecology and evolution
      </a>
    </h1>

    <span class="post-date">14 Jun 2012</span>

    <p>Many R packages/tools have come out recently for doing ecology and evolution. All of the below were described in Methods in Ecology and Evolution, except for spider, which came out in <a href="http://onlinelibrary.wiley.com/journal/10.1111/(ISSN)1755-0998">Molecular Ecology Resources</a>. Here are some highlights.</p>

<ul>
<li>mvabund <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00190.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/mvabund/index.html">get R pkg</a>

<ul>
<li>Model-based analysis of multivariate abundance data. Visualising data, fitting predictive models, checking assumptions, hypothesis testing.</li>
</ul>
</li>
<li>popdemo <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00222.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/popdemo/index.html">get R pkg</a>

<ul>
<li>Population demography using projection matrix analysis.</li>
</ul>
</li>
<li>motmot <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00132.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/motmot/index.html">get R pkg</a>

<ul>
<li>Models of trait macroevolution on trees</li>
</ul>
</li>
<li>spider <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2011.03108.x/abstract?deniedAccessCustomisedMessage=&amp;userIsAuthenticated=false">paper</a> - <a href="http://cran.r-project.org/web/packages/spider/index.html">get R pkg</a>

<ul>
<li>Analysis of species identity and evolution, with particular reference to DNA barcoding</li>
</ul>
</li>
<li>BaSTA <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2012.00186.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/BaSTA/index.html">get R pkg</a>

<ul>
<li>Bayesian estimation of age-specific survival from incomplete markâ€“recapture/recovery data with covariates</li>
</ul>
</li>
<li>abc <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00179.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/abc/index.html">get R pkg</a>

<ul>
<li>Approximate Bayesian Computation (ABC)</li>
</ul>
</li>
<li>RNetLogo <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00180.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/RNetLogo/index.html">get R pkg</a>

<ul>
<li>Running and exploring individual-based models implemented in NetLogo</li>
</ul>
</li>
<li>phytools <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00169.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/phytools/index.html">get R pkg</a>

<ul>
<li>Tools for phylogenetic comparative biology</li>
</ul>
</li>
<li>smatr <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00153.x/abstract">paper</a> - <a href="http://cran.r-project.org/web/packages/smatr/index.html">get R pkg</a>

<ul>
<li>Estimation and inference about allometric lines</li>
</ul>
</li>
<li>RBrownie <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00112.x/abstract">paper</a> - <a href="http://www.brianomeara.info/tutorials/brownie">get R pkg ?</a>

<ul>
<li>Testing hypotheses about rates of evolutionary change</li>
</ul>
</li>
<li>polytomy resolver <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00103.x/abstract">paper</a> - <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00103.x/suppinfo">get R pkg</a>

<ul>
<li>Resolve polytomies on dated phylogenies with their R scripts [here][].</li>
</ul>
</li>
</ul>


<p>And a cool tool came out for the Python programming language.</p>

<ul>
<li>NichePy <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.2041-210X.2011.00184.x/abstract">paper</a> - <a href="https://github.com/bastodian/NichePy">get python</a>

<ul>
<li>Modular tools for estimating the similarity of ecological niche and species distribution models</li>
</ul>
</li>
</ul>


  </div>
  
</div>

<!-- Pagination links -->
<div class="pagination">
  
    <a href="/page22" class="older">Older</a>
  
  
    
      <a href="/page20" class="newer">Newer</a>
    
  
</div>

    </div>

  </body>

  <footer>  
  <!-- Disqus code -->
  <script type="text/javascript">
      /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
      var disqus_shortname = 'recology'; // required: replace example with your forum shortname

      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function () {
          var s = document.createElement('script'); s.async = true;
          s.type = 'text/javascript';
          s.src = 'http://' + disqus_shortname + '.disqus.com/count.js';
          (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
      }());
  </script>

  <!-- Gaug.es Analytics -->
  <script type="text/javascript">
    var _gauges = _gauges || [];
    (function() {
      var t   = document.createElement('script');
      t.type  = 'text/javascript';
      t.async = true;
      t.id    = 'gauges-tracker';
      t.setAttribute('data-site-id', '4efd83a6f5a1f5158a000004');
      t.src = '//secure.gaug.es/track.js';
      var s = document.getElementsByTagName('script')[0];
      s.parentNode.insertBefore(t, s);
    })();
  </script>
</footer>
</html>
