<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>Datasets | Recology</title>
<meta name="keywords" content="">
<meta name="description" content="">
<meta name="author" content="">
<link rel="canonical" href="http://localhost:1313/tags/datasets/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.a4ec0cdfac731fabf363c0b5d5ab206254bbf9b6aaf985d695c6bb534196df16.css" integrity="sha256-pOwM36xzH6vzY8C11asgYlS7&#43;baq&#43;YXWlca7U0GW3xY=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" type="application/rss+xml" href="http://localhost:1313/tags/datasets/index.xml">
<link rel="alternate" hreflang="en" href="http://localhost:1313/tags/datasets/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="list" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Recology (Alt + H)">Recology</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/" title="Home">
                    <span>Home</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/index.xml" title="RSS">
                    <span>RSS</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main"> 
<header class="page-header">
  <h1>
    Datasets
  </h1>
</header>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">oai - an OAI-PMH client
    </h2>
  </header>
  <div class="entry-content">
    <p>oai is a general purpose client to work with any ‘OAI-PMH’ service. The ‘OAI-PMH’ protocol is described at https://www.openarchives.org/OAI/openarchivesprotocol.html. The main functions follow the OAI-PMH verbs:
GetRecord Identify ListIdentifiers ListMetadataFormats ListRecords ListSets The repo is at https://github.com/sckott/oai
I will be using this in a number of packages I maintain that use OAI-PMH data services. If you try it, let me know what you think.
This package is heading to rOpenSci soon: https://github.com/ropensci/onboarding/issues/19
Here’s a few usage examples:
...</p>
  </div>
  <footer class="entry-footer"><span title='2015-09-11 00:00:00 +0000 UTC'>September 11, 2015</span>&nbsp;·&nbsp;3 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to oai - an OAI-PMH client" href="http://localhost:1313/2015/09/oai-client/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Weecology can has new mammal dataset
    </h2>
  </header>
  <div class="entry-content">
    <p>So the Weecology folks have published a large dataset on mammal communities in a data paper in Ecology. I know nothing about mammal communities, but that doesn’t mean one can’t play with the data…
Their dataset consists of five csv files: communities, references, sites, species, and trapping data Where are these sites, and by the way, do they vary much in altitude? Let’s zoom in on just the states ...</p>
  </div>
  <footer class="entry-footer"><span title='2011-12-29 08:11:00 -0600 -0600'>December 29, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Weecology can has new mammal dataset" href="http://localhost:1313/2011/12/weecology-can-has-new-mammal-dataset/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">My little presentation on getting web data through R
    </h2>
  </header>
  <div class="entry-content">
    <p>With examples from rOpenSci R packages. p.s. I am no expert at this...
Web data from R View more presentations from schamber </p>
  </div>
  <footer class="entry-footer"><span title='2011-10-28 11:16:00.003 -0500 -0500'>October 28, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to My little presentation on getting web data through R" href="http://localhost:1313/2011/10/my-little-presentation-on-getting-web/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">rnpn: An R interface for the National Phenology Network
    </h2>
  </header>
  <div class="entry-content">
    <p>The team at rOpenSci and I have been working on a wrapper for the USA National Phenology Network API. The following is a demo of some of the current possibilities. We will have more functions down the road. Get the publicly available code, and contribute, at Github here. If you try this out look at the Description file for the required R packages to run rnpn. Let us know at Github (here) or at our website http://ropensci.org/, or in the comments below, or on twitter (@rOpenSci), what use cases you would like to see with the rnpn package.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-08-31 10:26:00 -0500 -0500'>August 31, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to rnpn: An R interface for the National Phenology Network" href="http://localhost:1313/2011/08/rnpn-r-interface-for-national-phenology/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">(#ESA11) rOpenSci: a collaborative effort to develop R-based tools for facilitating Open Science
    </h2>
  </header>
  <div class="entry-content">
    <p>Our development team would like to announce the launch of rOpenSci. As the title states, this project aims to create R packages to make open science more available to researchers.
http://ropensci.org/
What this means is that we seek to connect researchers using R with as much open data as possible, mainly through APIs. There are a number of R packages that already do this (e.g., infochimps, twitteR), but we are making more packages, e.g., for Mendeley, PLoS Journals, and taxonomic sources (ITIS, EOL, TNRS, Phylomatic, UBio).
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-08-08 10:06:00 -0500 -0500'>August 8, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to (#ESA11) rOpenSci: a collaborative effort to develop R-based tools for facilitating Open Science" href="http://localhost:1313/2011/08/esa11-ropensci-collaborative-effort-to_08/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Archiving ecology/evolution data sets online
    </h2>
  </header>
  <div class="entry-content">
    <p>We now have many options for archiving data sets online:
Dryad, KNB, Ecological Archives, Ecology Data Papers, Ecological Data, etc.
However, these portals largely do not communicate with one another as far as I know, and there is no way to search over all data set sources, again, as far as I know. So, I wonder if it would ease finding of all these different data sets to get these different sites to get their data sets cloned on a site like Infochimps, or have links from Infochimps. Infochimps already has APIs (and there’s an R wrapper for the Infochimps API already set up here: http://cran.r-project.org/web/packages/infochimps/index.html by Drew Conway), and they have discussions set up there, etc.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-07-15 21:02:00 -0500 -0500'>July 15, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Archiving ecology/evolution data sets online" href="http://localhost:1313/2011/07/archiving-ecologyevolution-data-sets/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">rbold: An R Interface for Bold Systems barcode repository
    </h2>
  </header>
  <div class="entry-content">
    <p>Have you ever wanted to search and fetch barcode data from Bold Systems?
I am developing functions to interface with Bold from R. I just started, but hopefully folks will find it useful.
The code is at Github here. The two functions are still very buggy, so please bring up issues below, or in the Issues area on Github. For example, some searches work and other similar searches don’t. Apologies in advance for the bugs.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-06-28 09:27:00.001 -0500 -0500'>June 28, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to rbold: An R Interface for Bold Systems barcode repository" href="http://localhost:1313/2011/06/rbold-an-r-interface-for-bold-systems-barcode-repository/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">iEvoBio 2011 Synopsis
    </h2>
  </header>
  <div class="entry-content">
    <p>We just wrapped up the 2011 iEvoBio meeting. It was awesome! If you didn’t go this year or last year, definitely think about going next year.
Here is a list of the cool projects that were discussed at the meeting (apologies if I left some out):
Vistrails: workflow tool, awesome project by Claudio Silva Commplish: purpose is to use via API’s, not with the web UI Phylopic: a database of life-form silouhettes, including an API for remote access, sweet! Gloome MappingLife: awesome geographic/etc data visualization interace on the web SuiteSMA: visualizating multiple alignments treeBASE: R interface to treebase, by Carl Boettiger VertNet: database for vertebrate natural history collections RevBayes: revamp of MrBayes, with GUI, etc. Phenoscape Knowledge Base Peter Midford lightning talk: talked about matching taxonomic and genetic data BiSciCol: biological science collections tracker Ontogrator TNRS: taxonomic name resolution service Barcode of Life data systems, and remote access Moorea Biocode Project Microbial LTER’s data BirdVis: interactive bird data visualization (Claudio Silva in collaboration with Cornell Lab of Ornithology) Crowdlabs: I think the site is down right now, another project by Claudio Silva Phycas: Bayesian phylogenetics, can you just call this from R? RIP MrBayes!!!! replaced by RevBayes (see 9 above) Slides of presentations will be at Slideshare (not all presentations up yet) A birds of a feather group I was involved in proposed an idea (TOL-o-matic) like Phylomatic, but of broader scope, for easy access and submission of trees, and perhaps even social (think just pushing a ‘SHARE’ button within PAUP, RevBayes, or other phylogenetics software)! Synopses of Birds of a Feather discussion groups: http://piratepad.net/iEvoBio11-BoF-reportouts </p>
  </div>
  <footer class="entry-footer"><span title='2011-06-22 21:38:00.001 -0500 -0500'>June 22, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to iEvoBio 2011 Synopsis" href="http://localhost:1313/2011/06/ievobio-2011-synopsis/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">OpenStates from R via API: watch your elected representatives
    </h2>
  </header>
  <div class="entry-content">
    <p>I am writing some functions to acquire data from the OpenStates project, via their API. They have a great support community at Google Groups as well.
On its face this post is not obviously about ecology or evolution, but well, our elected representatives do, so to speak, hold our environment in a noose, ready to let the Earth hang any day.
Code I am developing is over at Github. Here is an example of its use in R, in this case using the Bill Search option (billsearch.R on my Github site), and in this case you do not provide your API key in the function call, but instead put it in your .Rprofile file, which is called when you open R.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-06-10 23:19:00 -0500 -0500'>June 10, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to OpenStates from R via API: watch your elected representatives" href="http://localhost:1313/2011/06/openstates-from-r-via-api-watch-your-elected-representatives/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Farmer&#39;s markets data
    </h2>
  </header>
  <div class="entry-content">
    <p>I combined USDA data on farmer’s markets in the US with population census data to get an idea of the disparity in farmers markets by state, and then also expressed per capita.
Download USDA data here. The formatted file I used below is here (in excel format, although I read into R as csv file). The census data is read from url as below.
California has a ton of absolute number of farmer’s markets, but Vermont takes the cake by far with number of markets per capita. Iowa comes in a distant second behind Vermont in markets per capita.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-02-16 22:41:00.001 -0600 -0600'>February 16, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Farmer&#39;s markets data" href="http://localhost:1313/2011/02/farmer-s-markets-data/"></a>
</article>
<footer class="page-footer">
  <nav class="pagination">
    <a class="next" href="http://localhost:1313/tags/datasets/page/2/">Next&nbsp;&nbsp;»
    </a>
  </nav>
</footer>
    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Recology</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
