<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>Posts | Recology</title>
<meta name="keywords" content="">
<meta name="description" content="Posts - Recology">
<meta name="author" content="">
<link rel="canonical" href="http://localhost:1313/posts/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.a4ec0cdfac731fabf363c0b5d5ab206254bbf9b6aaf985d695c6bb534196df16.css" integrity="sha256-pOwM36xzH6vzY8C11asgYlS7&#43;baq&#43;YXWlca7U0GW3xY=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" type="application/rss+xml" href="http://localhost:1313/posts/index.xml">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="list" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Recology (Alt + H)">Recology</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/" title="Home">
                    <span>Home</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/index.xml" title="RSS">
                    <span>RSS</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main"> 
<header class="page-header">
  <h1>
    Posts
  </h1>
</header>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">RHIPE package in R for interfacing between Hadoop and R
    </h2>
  </header>
  <div class="entry-content">
    <p>RHIPE: An Interface Between Hadoop and R
Presented by Saptarshi Guha
And this review of methods for interfacing with Hadoop suggests R’s RHIPE is quite nice.
</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-04 19:11:00 -0500 -0500'>May 4, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to RHIPE package in R for interfacing between Hadoop and R" href="http://localhost:1313/2011/05/rhipe-package-in-r-for-interfacing-between-hadoop-and-r/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Treebase trees from R
    </h2>
  </header>
  <div class="entry-content">
    <p>UPDATE: See Carl Boettiger’s functions/package at Github for searching Treebase here.
Treebase is a great resource for phylogenetic trees, and has a nice interface for searching for certain types of trees. However, if you want to simply download a lot of trees for analyses (like that in Davies et al.), then you want to be able to access trees in bulk (I believe Treebase folks are working on an API though). I wrote some simple code for extracting trees from Treebase.org.It reads an xml file of (in this case consensus) URL’s for each tree, parses the xml, makes a vector of URL’s, reads the nexus files with error checking, remove trees that gave errors, then a simple plot looking at metrics of the trees.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-03 10:00:00.001 -0500 -0500'>May 3, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Treebase trees from R" href="http://localhost:1313/2011/05/treebase-trees-from-r/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Processing nested lists
    </h2>
  </header>
  <div class="entry-content">
    <p>So perhaps you have all figured this out already, but I was excited to figure out how to finally neatly get all the data frames, lists, vectors, etc. out of a nested list. It is as easy as nesting calls to the apply family of functions, in the case below, using plyr’s apply like functions. Take this example:
# Nested lists code, an example # Make a nested list mylist &lt;- list() mylist_ &lt;- list() for(i in 1:5) { for(j in 1:5) { mylist[[j]] &lt;- i*j } mylist_[[i]] &lt;- mylist } # return values from first part of list laply(mylist_[[1]], identity) [1] 1 2 3 4 5 # return all values laply(mylist_, function(x) laply(x, identity)) 1 2 3 4 5 [1,] 1 2 3 4 5 [2,] 2 4 6 8 10 [3,] 3 6 9 12 15 [4,] 4 8 12 16 20 [5,] 5 10 15 20 25 # perform some function, in this case sqrt of each value laply(mylist_, function(x) laply(x, function(x) sqrt(x))) 1 2 3 4 5 [1,] 1.000000 1.414214 1.732051 2.000000 2.236068 [2,] 1.414214 2.000000 2.449490 2.828427 3.162278 [3,] 1.732051 2.449490 3.000000 3.464102 3.872983 [4,] 2.000000 2.828427 3.464102 4.000000 4.472136 [5,] 2.236068 3.162278 3.872983 4.472136 5.000000 </p>
  </div>
  <footer class="entry-footer"><span title='2011-04-28 17:41:00 -0500 -0500'>April 28, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Processing nested lists" href="http://localhost:1313/2011/04/processing-nested-lists/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Running Phylip&#39;s contrast application for trait pairs from R
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is some code to run Phylip’s contrast application from R and get the output within R to easily manipulate yourself. Importantly, the code is written specifically for trait pairs only as the regular expression work in the code specifically grabs data from contast results when only two traits are input. You could easily change the code to do N traits. Note that the p-value calculated for the chi-square statistic is not output from contrast, but is calculated within the function ‘PhylipWithinSpContr’. In the code below there are two functions that make a lot of busy work easier: ‘WritePhylip’ and ‘PhylipWithinSpContr’. The first function is nice because the formatting required for data input to Phylip programs is so, well, awkward - and this function does it for you. The second function runs contrast and retrieves the output data. The example data set I produce in the code below has multiple individuals per species, so that contrasts are calculated taking into account within species variation. Get Phylip’s contrast documentation here.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-26 07:40:00.004 -0500 -0500'>April 26, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Running Phylip&#39;s contrast application for trait pairs from R" href="http://localhost:1313/2011/04/running-phylip-s-contrast-application-for-trait-pairs-from-r/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R: Randomization via Tip Shuffle
    </h2>
  </header>
  <div class="entry-content">
    <p>—UPDATE: I am now using code formatting from gist.github, so I replaced the old prettyR code (sorry guys). The github way is much easier and prettier. I hope readers like the change.
I wrote earlier about some code I wrote for running Phylometa (software to do phylogenetic meta-analysis) from R.
I have been concerned about what exactly is the right penalty for including phylogeny in a meta-analysis. E.g.: AIC is calculated from Q in Phylometa, and Q increases with tree size.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-16 13:44:00.004 -0500 -0500'>April 16, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R: Randomization via Tip Shuffle" href="http://localhost:1313/2011/04/phylometa-from-r-randomization-via-tip-shuffle/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">RStudio Beta 2 is Out!
    </h2>
  </header>
  <div class="entry-content">
    <p>RStudio Beta 2 (v0.93) « RStudio Blog
A new beta version of RStudio is out!
</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-11 09:33:00 -0500 -0500'>April 11, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to RStudio Beta 2 is Out!" href="http://localhost:1313/2011/04/rstudio-beta-2-is-out-/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Adjust branch lengths with node ages: comparison of two methods
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is an approach for comparing two methods of adjusting branch lengths on trees: bladj in the program Phylocom and a fxn written by Gene Hunt at the Smithsonian.
Get the code and example files (tree and node ages) at https://gist.github.com/938313
Get phylocom at http://www.phylodiversity.net/phylocom/
Gene Hunt’s method has many options you can mess with, including setting tip ages (not available in bladj), setting node ages, and minimum branch length imposed. You will notice that Gene’s method may be not the appropriate if you only have extant taxa.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-10 13:02:00.003 -0500 -0500'>April 10, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Adjust branch lengths with node ages: comparison of two methods" href="http://localhost:1313/2011/04/adjust-branch-lengths-with-node-ages-comparison-of-two-methods/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R - UDPATE
    </h2>
  </header>
  <div class="entry-content">
    <p>A while back I posted some messy code to run Phylometa from R, especially useful for processing the output data from Phylometa which is not easily done. The code is still quite messy, but it should work now. I have run the code with tens of different data sets and phylogenies so it should work.
I fixed errors when parentheses came up against numbers in the output, and other things. You can use the code for up to 4 levels of your grouping variable. In addition, there are some lines of code to plot the effect sizes with confidence intervals, comparing random and fixed effects models and phylogenetic and traditional models.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-01 16:18:00.003 -0500 -0500'>April 1, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R - UDPATE" href="http://localhost:1313/2011/04/phylometa-from-r-udpate/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Bio-ORACLE
    </h2>
  </header>
  <div class="entry-content">
    <p>Bio-ORACLE
A new dataset available of geophysical, biotic and climate data. Should be fun to play with in R.
</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-25 08:38:00 -0500 -0500'>March 25, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Bio-ORACLE" href="http://localhost:1313/2011/03/bio-oracle/"></a>
</article>

<article class="post-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">basic ggplot2 network graphs ver2
    </h2>
  </header>
  <div class="entry-content">
    <p>I posted last week a simple function to plot networks using ggplot2 package. Here is version 2. I still need to work on figuring out efficient vertex placement.
Changes in version 2:
You have one of three options: use an igraph object, a matrix, or a dataframe (matrices will be converted to data frames within the function) If you have data on food webs similar to that provided in the Takapoto dataset provided in the NetIndices package, you can set trophic = “TRUE”, and gggraph will use the function TrophInd to assign trophic levels (the y axis value) to each vertex/node. You have to provide additional information along with this option such as what the imports and exports are, see NetIndices documentation. I added some simple error checking. if using method=“df” and trophic=“FALSE”, x axis placement of vertices is now done using the function degreex (see inside the fxn), which sorts vertices according to their degree (so the least connected species are on the left of the graph; note that species with the same degree are not stacked on the y-axis because e.g., two vertices of degree=5 would get x=3 then x=4). # ggraph Version 2 require(bipartite) require(igraph) require(ggplot2) # gggraph, version 3 g = an igraph graph object, a matrix, or data frame # vplace = type of vertex placement assignment, one of rnorm, runif, etc. # method = one of &#39;df&#39; for data frame, &#39;mat&#39; for matrix or &#39;igraph&#39; for an # igraph graph object trophic = TRUE or FALSE for using Netindices # function TrophInd to determine trophic level (y value in graph) # trophinames = columns in matrix or dataframe to use for calculating # trophic level import = named or refereced by col# columns of matrix or # dataframe to use for import argument of TrophInd export = named or # refereced by col# columns of matrix or dataframe to use for export # argument of TrophInd dead = named or refereced by col# columns of matrix # or dataframe to use for dead argument of TrophInd gggraph &lt;- function(g, vplace = rnorm, method, trophic = &#34;FALSE&#34;, trophinames, import, export) { degreex &lt;- function(x) { degreecol &lt;- apply(x, 2, function(y) length(y[y &gt; 0])) degreerow &lt;- apply(x, 1, function(y) length(y[y &gt; 0])) degrees &lt;- sort(c(degreecol, degreerow)) df &lt;- data.frame(degrees, x = seq(1, length(degrees), 1)) df$value &lt;- rownames(df) df } # require igraph if (!require(igraph)) stop(&#34;must first install &#39;igraph&#39; package.&#34;) # require ggplot2 if (!require(ggplot2)) stop(&#34;must first install &#39;ggplot2&#39; package.&#34;) if (method == &#34;df&#34;) { if (class(g) == &#34;matrix&#34;) { g &lt;- as.data.frame(g) } if (class(g) != &#34;data.frame&#34;) stop(&#34;object must be of class &#39;data.frame.&#39;&#34;) if (trophic == &#34;FALSE&#34;) { # data preparation from adjacency matrix temp &lt;- data.frame(expand.grid(dimnames(g))[1:2], as.vector(as.matrix(g))) temp &lt;- temp[(temp[, 3] &gt; 0) &amp; !is.na(temp[, 3]), ] temp &lt;- temp[sort.list(temp[, 1]), ] g_df &lt;- data.frame(rows = temp[, 1], cols = temp[, 2], freqint = temp[, 3]) g_df$id &lt;- 1:length(g_df[, 1]) g_df &lt;- data.frame(id = g_df[, 4], rows = g_df[, 1], cols = g_df[, 2], freqint = g_df[, 3]) g_df_ &lt;- melt(g_df, id = c(1, 4)) xy_s &lt;- data.frame(degreex(g), y = rnorm(length(unique(g_df_$value)))) g_df_2 &lt;- merge(g_df_, xy_s, by = &#34;value&#34;) } else if (trophic == &#34;TRUE&#34;) { # require NetIndices if (!require(NetIndices)) stop(&#34;must first install &#39;NetIndices&#39; package.&#34;) # data preparation from adjacency matrix temp &lt;- data.frame(expand.grid(dimnames(g[-trophinames, -trophinames]))[1:2], as.vector(as.matrix(g[-trophinames, -trophinames]))) temp &lt;- temp[(temp[, 3] &gt; 0) &amp; !is.na(temp[, 3]), ] temp &lt;- temp[sort.list(temp[, 1]), ] g_df &lt;- data.frame(rows = temp[, 1], cols = temp[, 2], freqint = temp[, 3]) g_df$id &lt;- 1:length(g_df[, 1]) g_df &lt;- data.frame(id = g_df[, 4], rows = g_df[, 1], cols = g_df[, 2], freqint = g_df[, 3]) g_df_ &lt;- melt(g_df, id = c(1, 4)) xy_s &lt;- data.frame(value = unique(g_df_$value), x = rnorm(length(unique(g_df_$value))), y = TrophInd(g, Import = import, Export = export)[, 1]) g_df_2 &lt;- merge(g_df_, xy_s, by = &#34;value&#34;) } # plotting p &lt;- ggplot(g_df_2, aes(x, y)) &#43; geom_point(size = 5) &#43; geom_line(aes(size = freqint, group = id)) &#43; geom_text(size = 3, hjust = 1.5, aes(label = value)) &#43; theme_bw() &#43; opts(panel.grid.major = theme_blank(), panel.grid.minor = theme_blank(), axis.text.x = theme_blank(), axis.text.y = theme_blank(), axis.title.x = theme_blank(), axis.title.y = theme_blank(), axis.ticks = theme_blank(), panel.border = theme_blank(), legend.position = &#34;none&#34;) p # return graph } else if (method == &#34;igraph&#34;) { if (class(g) != &#34;igraph&#34;) stop(&#34;object must be of class &#39;igraph.&#39;&#34;) # data preparation from igraph object g_ &lt;- get.edgelist(g) g_df &lt;- as.data.frame(g_) g_df$id &lt;- 1:length(g_df[, 1]) g_df &lt;- melt(g_df, id = 3) xy_s &lt;- data.frame(value = unique(g_df$value), x = vplace(length(unique(g_df$value))), y = vplace(length(unique(g_df$value)))) g_df2 &lt;- merge(g_df, xy_s, by = &#34;value&#34;) # plotting p &lt;- ggplot(g_df2, aes(x, y)) &#43; geom_point(size = 2) &#43; geom_line(size = 0.3, aes(group = id, linetype = id)) &#43; geom_text(size = 3, hjust = 1.5, aes(label = value)) &#43; theme_bw() &#43; opts(panel.grid.major = theme_blank(), panel.grid.minor = theme_blank(), axis.text.x = theme_blank(), axis.text.y = theme_blank(), axis.title.x = theme_blank(), axis.title.y = theme_blank(), axis.ticks = theme_blank(), panel.border = theme_blank(), legend.position = &#34;none&#34;) p # return graph } else stop(paste(&#34;do not recognize method = \&#34;&#34;, method, &#34;\&#34;;\nmethods are \&#34;df\&#34; and \&#34;igraph\&#34;&#34;, sep = &#34;&#34;)) } # Eg library(NetIndices) data(Takapoto) gggraph(Takapoto, vplace = rnorm, method = &#34;df&#34;, trophic = &#34;TRUE&#34;, trophinames = c(8:10), import = &#34;CO2&#34;, export = c(&#34;CO2&#34;, &#34;Sedimentation&#34;, &#34;Grazing&#34;)) ...</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-23 14:09:00 -0500 -0500'>March 23, 2011</span>&nbsp;·&nbsp;5 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to basic ggplot2 network graphs ver2" href="http://localhost:1313/2011/03/basic-ggplot2-network-graphs-ver2/"></a>
</article>
<footer class="page-footer">
  <nav class="pagination">
    <a class="prev" href="http://localhost:1313/posts/page/20/">
      «&nbsp;Prev&nbsp;
    </a>
    <a class="next" href="http://localhost:1313/posts/page/22/">Next&nbsp;&nbsp;»
    </a>
  </nav>
</footer>
    </main>
    
<footer class="footer">
        <span>&copy; 2025 <a href="http://localhost:1313/">Recology</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
