<!DOCTYPE html>

<html>
  <head>

    <meta http-equiv="Content-type" content="text/html;charset=UTF-8">

    <title>Recology: An exploration of using R for ecology, evolution, and open science.</title>

    <script type="text/javascript" src="http://use.typekit.com/jjw2jrj.js"></script>
    <script type="text/javascript">try{Typekit.load();}catch(e){}</script>

    <link type="text/css" rel="stylesheet" href="/stylesheets/reset.css">
    <link type="text/css" rel="stylesheet" href="/stylesheets/screen.css">
    <link type="text/css" rel="stylesheet" href="/stylesheets/code.css">
    <link rel="alternate" type="application/atom+xml"
      title="User Obsessed Feed" href="http://feeds.feedburner.com/userobsessed" />

    <link rel="apple-touch-icon" href="/images/iOS-114.png" />
    <link rel="apple-touch-icon-precomposed" href="/images/iOS-57.png" />
    <link rel="apple-touch-icon-precomposed" sizes="72x72" media="screen and (resolution: 132dpi)" href="/images/iOS-72.png" />
    <link rel="apple-touch-icon-precomposed" sizes="114x114" media="screen and (resolution: 326dpi)" href="/images/iOS-114.png" />

    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0" />



  </head>
  <body class="default">

    <section id="main">

      <section id="masthead" class="group">
        <header class="group">
          <p><a href="/">Recology</a></p>
          <h1>An exploration of using R for ecology, evolution, and open science.</h1>
          <img src="/images/uo-logo.png">
        </header>
      </section>

      <section id="focus" class="group">

        <section id="content" class="group">
          <section id="posts">
  <ol>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/05/04/rhipe-package-in-r-for-interfacing-between-hadoop-and-r/">RHIPE package in R for interfacing between Hadoop and R</a></h2>
  </header>
  
    <div><strong><a href="http://www.lecturemaker.com/2011/02/rhipe/#video" title="Click link to go to the video page">RHIPE: An Interface Between Hadoop and R</a></strong><br />Presented by Saptarshi Guha</div><a href="http://www.lecturemaker.com/2011/02/rhipe/#video"><img alt="Video Link" border="0" height="169" src="http://www.lecturemaker.com/lectures/RMeetUp2010/RHIPE_Lecture.jpg" title="Click image to go to the video page" width="300" /></a>          <br /><br /><br />And <a href="http://blog.piccolboni.info/2011/04/looking-for-map-reduce-language.html">this review</a> of methods for interfacing with Hadoop suggests R's RHIPE is quite nice.

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-05-04T12:00:00-05:00">Wednesday, May  4</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F05%2F04%2Frhipe-package-in-r-for-interfacing-between-hadoop-and-r%2F&amp;title=RHIPE%20package%20in%20R%20for%20interfacing%20between%20Hadoop%20and%20R" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/05/03/treebase-trees-from-r/">Treebase trees from R</a></h2>
  </header>
  
    UPDATE: See Carl Boettiger's functions/package at Github for searching Treebase <a href="https://github.com/ropensci/treeBASE">here</a>.<br /><br /><br /><br />Treebase is a great resource for phylogenetic trees, and has a nice interface for searching for certain types of trees. However, if you want to simply download a lot of trees for analyses (like that in <a href="http://biology.mcgill.ca/faculty/davies/pdfs/Davies_etal_Evolution_2011.pdf">Davies et al.</a>), then you want to be able to access trees in bulk (I believe Treebase folks are working on an API though). I wrote some simple code for extracting trees from <a href="http://treebase.org/">Treebase.org</a>.<br /><br />It reads an xml file of (in this case consensus) URL's for each tree, parses the xml, makes a vector of URL's, reads the nexus files with error checking, remove trees that gave errors, then a simple plot looking at metrics of the trees.<br /><br />Is there an easier way to do this?<br /><br /><br /><br /><script src="https://gist.github.com/953468.js?file=treebase_code.R"></script><br /><div class="separator" style="clear: both; text-align: center;"><a href="http://2.bp.blogspot.com/-AaMexPVCreo/TcAW171ZBaI/AAAAAAAAEbc/bDafe7YgGcw/s1600/sampetreebaseplot.png" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;"><img border="0" height="256" src="http://2.bp.blogspot.com/-AaMexPVCreo/TcAW171ZBaI/AAAAAAAAEbc/bDafe7YgGcw/s320/sampetreebaseplot.png" width="320" /></a></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-05-03T12:00:00-05:00">Tuesday, May  3</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F05%2F03%2Ftreebase-trees-from-r%2F&amp;title=Treebase%20trees%20from%20R" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/28/processing-nested-lists/">Processing nested lists</a></h2>
  </header>
  
    So perhaps you have all figured this out already, but I was excited to figure out how to finally neatly get all the data frames, lists, vectors, etc. out of a nested list. It is as easy as nesting calls to the apply family of functions, in the case below, using plyr's apply like functions. Take this example:<br /><br /><br /><br /><div style="overflow: auto;"><div class="geshifilter"><pre class="r geshifilter-R"><span style="color: #666666; font-family: monospace; font-style: italic;"># Nested lists code, an example</span><br /><span style="color: #666666; font-family: monospace; font-style: italic;"># Make a nested list</span><br />mylist <span style="font-family: monospace;">&lt;-</span> <a href="http://inside-r.org/r-doc/base/list" style="font-family: monospace;"><span style="color: #003399; font-weight: bold;">list</span></a><span style="color: #009900; font-family: monospace;">(</span><span style="color: #009900; font-family: monospace;">)</span><br />mylist_ <span style="font-family: monospace;">&lt;-</span> <a href="http://inside-r.org/r-doc/base/list" style="font-family: monospace;"><span style="color: #003399; font-weight: bold;">list</span></a><span style="color: #009900; font-family: monospace;">(</span><span style="color: #009900; font-family: monospace;">)</span><br /><span style="color: black; font-family: monospace; font-weight: bold;">for</span><span style="color: #009900; font-family: monospace;">(</span>i <span style="color: black; font-family: monospace; font-weight: bold;">in</span> <span style="color: #cc66cc; font-family: monospace;">1</span><span style="font-family: monospace;">:</span><span style="color: #cc66cc; font-family: monospace;">5</span><span style="color: #009900; font-family: monospace;">)</span> <span style="color: #009900; font-family: monospace;">{</span><br /> <span style="color: black; font-family: monospace; font-weight: bold;">for</span><span style="color: #009900; font-family: monospace;">(</span>j <span style="color: black; font-family: monospace; font-weight: bold;">in</span> <span style="color: #cc66cc; font-family: monospace;">1</span><span style="font-family: monospace;">:</span><span style="color: #cc66cc; font-family: monospace;">5</span><span style="color: #009900; font-family: monospace;">)</span> <span style="color: #009900; font-family: monospace;">{</span><br />  mylist<span style="color: #009900; font-family: monospace;">[</span><span style="color: #009900; font-family: monospace;">[</span>j<span style="color: #009900; font-family: monospace;">]</span><span style="color: #009900; font-family: monospace;">]</span> <span style="font-family: monospace;">&lt;-</span> i<span style="font-family: monospace;">*</span>j<br /> <span style="color: #009900; font-family: monospace;">}</span><br />mylist_<span style="color: #009900; font-family: monospace;">[</span><span style="color: #009900; font-family: monospace;">[</span>i<span style="color: #009900; font-family: monospace;">]</span><span style="color: #009900; font-family: monospace;">]</span> <span style="font-family: monospace;">&lt;-</span> mylist<br /><span style="color: #009900; font-family: monospace;">}</span><br />&nbsp;<br /><span style="color: #666666; font-family: monospace; font-style: italic;"># return values from first part of list</span><br />laply<span style="color: #009900; font-family: monospace;">(</span>mylist_<span style="color: #009900; font-family: monospace;">[</span><span style="color: #009900; font-family: monospace;">[</span><span style="color: #cc66cc; font-family: monospace;">1</span><span style="color: #009900; font-family: monospace;">]</span><span style="color: #009900; font-family: monospace;">]</span><span style="color: #339933; font-family: monospace;">,</span> <a href="http://inside-r.org/r-doc/base/identity" style="font-family: monospace;"><span style="color: #003399; font-weight: bold;">identity</span></a><span style="color: #009900; font-family: monospace;">)</span><br />[1] 1 2 3 4 5</pre><pre class="r geshifilter-R">&nbsp;<br /><span style="color: #666666; font-family: monospace; font-style: italic;"># return all values</span><br />laply<span style="color: #009900; font-family: monospace;">(</span>mylist_<span style="color: #339933; font-family: monospace;">,</span> <a href="http://inside-r.org/r-doc/base/function" style="font-family: monospace;"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900; font-family: monospace;">(</span>x<span style="color: #009900; font-family: monospace;">)</span> laply<span style="color: #009900; font-family: monospace;">(</span>x<span style="color: #339933; font-family: monospace;">,</span> <a href="http://inside-r.org/r-doc/base/identity" style="font-family: monospace;"><span style="color: #003399; font-weight: bold;">identity</span></a><span style="color: #009900; font-family: monospace;">)</span><span style="color: #009900; font-family: monospace;">)</span><br />     1  2  3  4  5<br />[1,] 1  2  3  4  5<br />[2,] 2  4  6  8 10<br />[3,] 3  6  9 12 15<br />[4,] 4  8 12 16 20<br />[5,] 5 10 15 20 25</pre><pre class="r geshifilter-R" style="font-family: monospace;">&nbsp;<br /><span style="color: #666666; font-style: italic;"># perform some function, in this case sqrt of each value</span><br />laply<span style="color: #009900;">(</span>mylist_<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>x<span style="color: #009900;">)</span> laply<span style="color: #009900;">(</span>x<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>x<span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/sqrt"><span style="color: #003399; font-weight: bold;">sqrt</span></a><span style="color: #009900;">(</span>x<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span></pre><pre class="r geshifilter-R"><span style="color: #009900;">  </span><br />        1        2        3        4        5<br />[1,] 1.000000 1.414214 1.732051 2.000000 2.236068<br />[2,] 1.414214 2.000000 2.449490 2.828427 3.162278<br />[3,] 1.732051 2.449490 3.000000 3.464102 3.872983<br />[4,] 2.000000 2.828427 3.464102 4.000000 4.472136<br />[5,] 2.236068 3.162278 3.872983 4.472136 5.000000<br /></pre></div></div><br /><br /><a href="http://www.inside-r.org/pretty-r" title="Created by Pretty R at inside-R.org">Created by Pretty R at inside-R.org</a>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-28T12:00:00-05:00">Thursday, April 28</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F28%2Fprocessing-nested-lists%2F&amp;title=Processing%20nested%20lists" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/26/running-phylip-s-contrast-application-for-trait-pairs-from-r/">Running Phylip's contrast application for trait pairs from R</a></h2>
  </header>
  
    Here is some code to run Phylip's contrast application from R and get the output within R to easily manipulate yourself. Importantly, the code is written specifically for trait pairs only as the regular expression work in the code specifically grabs data from contast results when only two traits are input. You could easily change the code to do N traits. Note that the p-value calculated for the chi-square statistic is not output from contrast, but is calculated within the function 'PhylipWithinSpContr'. In the code below there are two functions that make&nbsp;a lot of busy work easier: 'WritePhylip' and&nbsp;'PhylipWithinSpContr'. The first function is nice because the formatting required for data input to Phylip programs is so, well, awkward &nbsp;- and this function does it for you. The second function runs contrast and retrieves the output data. The example data set I produce in the code below has multiple individuals per species, so that contrasts are calculated taking into account within species variation. Get Phylip's contrast documentation <a href="http://evolution.genetics.washington.edu/phylip/doc/contrast.html">here</a>.<br /><br />Note that the data input format allows only 10 characters for the species name, so I suggest if your species names are longer than 10 characters use the function abbreviate, for example, to shorten all names to no longer than 10 characters. Also, within the function WritePhylip I concatenate species names and their number of individuals per species leaving plenty of space.<br /><br />Also, mess around with the options in the "system" call to get what you want. For example, I used "R", "W" and "Y", meaning replace old outfile (R), then turn on within species analyses (W), then accept all options (Y). E..g, if you don't have an old outfile, then you obviously don't need to replace the old file with the "R" command.<br /><br />(p.s. I have not tried this on a windows machine).<br /><br /><br /><script src="https://gist.github.com/942176.js?file=phylip_fromR.R"></script><br /><br />Here is example output:<br /><br /><span class="Apple-style-span" style="font-family: Monaco; font-size: 12px; line-height: 17px; white-space: pre-wrap;"></span><br /><pre class="G1dpdwhmFL" style="border-bottom-style: none; border-color: initial; border-left-style: none; border-right-style: none; border-top-style: none; border-width: initial; font-family: Monaco; font-size: 9pt !important; line-height: 1.45; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; margin-top: 0px; outline-color: initial; outline-style: none; outline-width: initial; white-space: pre-wrap !important;" tabindex="0"><span class="G1dpdwhmIL  ace_keyword" style="white-space: pre;">&gt; </span><span class="G1dpdwhmMK  ace_keyword">datout<br /></span>               names2   dat...1.    dat...2.<br />1      VarAIn_VarAest   0.000110   -0.000017<br />2      VarAIn_VarAest  -0.000017    0.000155<br />3      VarAIn_VarEest   0.790783   -0.063097<br />4      VarAIn_VarEest  -0.063097    0.981216<br />5      VarAIn_VarAreg   1.000000   -0.107200<br />6      VarAIn_VarAreg  -0.151800    1.000000<br />7     VarAIn_VarAcorr   1.000000   -0.127600<br />8     VarAIn_VarAcorr  -0.127600    1.000000<br />9      VarAIn_VarEreg   1.000000   -0.064300<br />10     VarAIn_VarEreg  -0.079800    1.000000<br />11    VarAIn_VarEcorr   1.000000   -0.071600<br />12    VarAIn_VarEcorr  -0.071600    1.000000<br />13    VarAOut_VarEest   0.790734   -0.063104<br />14    VarAOut_VarEest  -0.063104    0.981169<br />15    VarAOut_VarEreg   1.000000   -0.064300<br />16    VarAOut_VarEreg  -0.079800    1.000000<br />17   VarAOut_VarEcorr   1.000000   -0.071600<br />18   VarAOut_VarEcorr  -0.071600    1.000000<br />19    logL_withvar_df -68.779770    6.000000<br />20 logL_withoutvar_df -68.771450    3.000000<br />21           chisq_df  -0.016640    3.000000<br />22            chisq_p   1.000000 -999.000000</pre>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-26T12:00:00-05:00">Tuesday, April 26</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F26%2Frunning-phylip-s-contrast-application-for-trait-pairs-from-r%2F&amp;title=Running%20Phylip's%20contrast%20application%20for%20trait%20pairs%20from%20R" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/16/phylometa-from-r-randomization-via-tip-shuffle/">Phylometa from R: Randomization via Tip Shuffle</a></h2>
  </header>
  
    ---UPDATE: I am now using code formatting from gist.github, so I replaced the old prettyR code (sorry guys). The github way is much easier and prettier. I hope readers like the change.<br /><br /><br /><br /><br /><a href="http://r-ecology.blogspot.com/2011/04/phylometa-from-r-udpate.html">I wrote earlier</a> about some code I wrote for running Phylometa (software to do phylogenetic meta-analysis) from R.<br /><br />I have been concerned about what exactly is the right penalty for including phylogeny in a meta-analysis. E.g.: AIC is calculated from Q in Phylometa, and Q increases with tree size.<br /><br />So, I wrote some code to shuffle the tips of your tree N number of times, run Phylometa, and extract just the "Phylogenetic MA" part of the output. So, we compare the observed output (without tip shuffling) to the distribution of the tip shuffled output, and we can calculate a P-value from that. The code I wrote simply extracts the pooled effect size for fixed and also random-effects models. But you could change the code to extract whatever you like for the randomization.<br /><br />I think the point of this code is not to estimate your pooled effects, etc., but may be an alternative way to compare traditional to phylogenetic MA where hopefully simply incorporating a tree is not penalizing the meta-analysis so much&nbsp;that you will&nbsp;always accept the traditional MA as better.<br /><br />Get the code <a href="https://gist.github.com/925343#file_phylometa_rand_fxn_one.r">here</a>, and also below. Get the example <a href="http://wp.me/PRT1F-2R">tree file</a> and <a href="http://wp.me/PRT1F-2S">data file</a>, named "phylogeny.txt" and "metadata_2g.txt", respectively below (or use your own data!). You need the file "phylometa_fxn.r" from my website, get <a href="https://gist.github.com/939970">here</a>, but just call it using source as seen below.<br /><br /><script src="https://gist.github.com/925343.js?file=phylometa_rand_fxn_one.R"></script><br /><br />As you can see, the observed values fall well within the distribution of values obtained from shuffling tips. &nbsp;P-values were 0.64 and 0.68 for fixed- and random-effects MA's, respectively. This suggests, to me at least, that the traditional (distribution of tip shuffled analyses, the histograms below) and phylogenetic (red lines) MA's are not&nbsp;different. The way I would use this is as an additional analysis to the actual Phylometa output.<br /><div class="separator" style="clear: both; text-align: center;"><a href="http://4.bp.blogspot.com/-fpEjXUBvAw8/TanftVw49QI/AAAAAAAAEbY/z9rJKThxRMo/s1600/metadata_2g.txt.jpeg" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;"><img border="0" height="400" src="http://4.bp.blogspot.com/-fpEjXUBvAw8/TanftVw49QI/AAAAAAAAEbY/z9rJKThxRMo/s400/metadata_2g.txt.jpeg" width="400" /></a></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-16T12:00:00-05:00">Saturday, April 16</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F16%2Fphylometa-from-r-randomization-via-tip-shuffle%2F&amp;title=Phylometa%20from%20R%3A%20Randomization%20via%20Tip%20Shuffle" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/11/rstudio-beta-2-is-out-/">RStudio Beta 2 is Out!</a></h2>
  </header>
  
    <a href="http://blog.rstudio.org/2011/04/11/rstudio-beta2/">RStudio Beta 2 (v0.93) Â« RStudio Blog</a><div><br /></div><div>A new beta version of RStudio is out! </div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-11T12:00:00-05:00">Monday, April 11</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F11%2Frstudio-beta-2-is-out-%2F&amp;title=RStudio%20Beta%202%20is%20Out!" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/10/adjust-branch-lengths-with-node-ages-comparison-of-two-methods/">Adjust branch lengths with node ages: comparison of two methods</a></h2>
  </header>
  
    Here is an approach for comparing two methods of adjusting branch lengths on trees: bladj in the program Phylocom and a fxn written by Gene Hunt at the Smithsonian.<br /><br />Get the code and example files (tree and node ages) <a href="https://gist.github.com/938313">here</a><br />Get phylocom here:&nbsp;<a href="http://www.phylodiversity.net/phylocom/">http://www.phylodiversity.net/phylocom/</a><br /><br />Gene Hunt's method has many options you can mess with, including setting tip ages (not available in bladj), setting node ages, and minimum branch length imposed. You will notice that Gene's method may be not the appropriate if you only have extant taxa.<br /><br />The function AdjBrLens uses as input a newick tree file and a text file of node ages, and uses functions you can simply run by "source" the R file bladjing_twomethods.R file from <a href="https://gist.github.com/938313">here</a>.<br /><br />Note that blad does not like numbers for node names, so you have to put a character in front of a number of just character names for nodes.<br /><br /><br /><br /><div style="overflow: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;"><span style="color: #666666; font-style: italic;"># This is where the work happens... </span><br /><span style="color: #666666; font-style: italic;"># Directory below needs to have at least three items:</span><br /><span style="color: #666666; font-style: italic;">#  1. phylocom executable for windows or mac</span><br /><span style="color: #666666; font-style: italic;">#  2. tree newick file</span><br /><span style="color: #666666; font-style: italic;">#  3. node ages file as required by phylocom, see their manual</span><br /><span style="color: #666666; font-style: italic;"># Output: trees_out is a list of three trees, the original, bladj, and Gene Hunt's method</span><br /><span style="color: #666666; font-style: italic;"># Also, within the function all three trees are written to file as PDFs</span><br /><a href="http://inside-r.org/r-doc/base/setwd"><span style="color: #003399; font-weight: bold;">setwd</span></a><span style="color: #009900;">(</span><span style="color: blue;">"/Mac/R_stuff/Blog_etc/Bladjing"</span><span style="color: #009900;">)</span> <span style="color: #666666; font-style: italic;"># set working directory</span><br /><a href="http://inside-r.org/r-doc/base/source"><span style="color: #003399; font-weight: bold;">source</span></a><span style="color: #009900;">(</span><span style="color: blue;">"bladjing_twomethods.R"</span><span style="color: #009900;">)</span> <span style="color: #666666; font-style: italic;"># run functions from source file</span><br />trees_out &lt;- AdjBrLens<span style="color: #009900;">(</span><span style="color: blue;">"tree.txt"</span><span style="color: #339933;">,</span> <span style="color: blue;">"nodeages.txt"</span><span style="color: #009900;">)</span><br />&nbsp;<br /><span style="color: #666666; font-style: italic;"># plot trees of three methods together, </span><br /><span style="color: #666666; font-style: italic;"># with nodes with age estimates labeled</span><br /><a href="http://inside-r.org/r-doc/grDevices/jpeg"><span style="color: #003399; font-weight: bold;">jpeg</span></a><span style="color: #009900;">(</span><span style="color: blue;">"threeplots.jpeg"</span><span style="color: #339933;">,</span> quality=<span style="color: #cc66cc;">100</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/layout"><span style="color: #003399; font-weight: bold;">layout</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/matrix"><span style="color: #003399; font-weight: bold;">matrix</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span>:<span style="color: #cc66cc;">3</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/plot"><span style="color: #003399; font-weight: bold;">plot</span></a><span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br />nodelabels<span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span>$node.label<span style="color: #339933;">,</span> cex = <span style="color: #cc66cc;">0.6</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/title"><span style="color: #003399; font-weight: bold;">title</span></a><span style="color: #009900;">(</span><span style="color: blue;">"original tree"</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/plot"><span style="color: #003399; font-weight: bold;">plot</span></a><span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br />nodelabels<span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span>$node.label<span style="color: #339933;">,</span> cex = <span style="color: #cc66cc;">0.6</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/title"><span style="color: #003399; font-weight: bold;">title</span></a><span style="color: #009900;">(</span><span style="color: blue;">"bladj method"</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/plot"><span style="color: #003399; font-weight: bold;">plot</span></a><span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br />nodelabels<span style="color: #009900;">(</span>trees_out<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span>$node.label<span style="color: #339933;">,</span> cex = <span style="color: #cc66cc;">0.6</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/graphics/title"><span style="color: #003399; font-weight: bold;">title</span></a><span style="color: #009900;">(</span><span style="color: blue;">"gene hunt method, scalePhylo"</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/grDevices/dev.off"><span style="color: #003399; font-weight: bold;">dev.off</span></a><span style="color: #009900;">(</span><span style="color: #009900;">)</span></pre></div></div><a href="http://www.inside-r.org/pretty-r" title="Created by Pretty R at inside-R.org">Created by Pretty R at inside-R.org</a><br /><br /><br />It is sort of hard to see the differences in&nbsp;the branch length changes here, but the individual output trees will reveal the differences better.<br /><div class="separator" style="clear: both; text-align: center;"><a href="http://2.bp.blogspot.com/-tLK1y12TYlI/TaHwayCs3GI/AAAAAAAAEbU/rPsFYqSEDuI/s1600/threeplots.jpeg" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;"><img border="0" src="http://2.bp.blogspot.com/-tLK1y12TYlI/TaHwayCs3GI/AAAAAAAAEbU/rPsFYqSEDuI/s1600/threeplots.jpeg" /></a></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-10T12:00:00-05:00">Sunday, April 10</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F10%2Fadjust-branch-lengths-with-node-ages-comparison-of-two-methods%2F&amp;title=Adjust%20branch%20lengths%20with%20node%20ages%3A%20comparison%20of%20two%20methods" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/04/01/phylometa-from-r-udpate/">Phylometa from R - UDPATE</a></h2>
  </header>
  
    <a href="http://r-ecology.blogspot.com/2010/12/phylogenetic-meta-analysis-in-r-using.html">A while back I posted some messy code to run Phylometa from R</a>, especially useful for processing the output data from Phylometa which is not easily done. The code is still quite messy, but it should work now. I have run the code with tens of different data sets and phylogenies so it should work. <br /><div><br /></div><div>I fixed errors when parentheses came up against numbers in the output, and other things. You can use the code for up to 4 levels of your grouping variable. In addition, there are some lines of code to plot the effect sizes with confidence intervals, comparing random and fixed effects models and phylogenetic and traditional models.&nbsp;</div><div><br /></div><div>Get the code at my website:</div><div>-<span class="Apple-style-span" style="color: #333333; font-family: Georgia, 'Bitstream Charter', serif; font-size: 14px; line-height: 23px;"><a href="https://gist.github.com/939970" style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; color: #1c9bdc; font-style: inherit; font-weight: inherit; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; margin-top: 0px; outline-color: initial; outline-style: initial; outline-width: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; text-decoration: underline; vertical-align: baseline;" target="_blank">phylometa_fxn.R</a></span></div><div>-<span class="Apple-style-span" style="color: #333333; font-family: Georgia, 'Bitstream Charter', serif; font-size: 14px; line-height: 23px;"><a href="https://gist.github.com/939971" style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; color: #1c9bdc; font-style: inherit; font-weight: inherit; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; margin-top: 0px; outline-color: initial; outline-style: initial; outline-width: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; text-decoration: underline; vertical-align: baseline;" target="_blank">phylometa_inR_run.R</a></span></div><div>- Use the first file to do the work, calling the second file using source().<br />- This new code works with Marc's new version of Phylometa, so please update:&nbsp;http://lajeunesse.myweb.usf.edu/publications.html</div><div><br /></div><div>Again, please let me know if it doesn't work, if it's worthless, what changes could make it better.<br /><br />Some notes on tree formatting for Phylometa. <br />1.&nbsp;Trees cannot have node labels - remove them (e.g., tree$node.label &lt; NULL).<br />2. Trees cannot have zero length branches. This may seem like a non-problem, but it might be for example if you have resolved polytomies and zero length branches are added to resolve the polytomy.<br />3. I think you cannot have a branch length on the root branch.<br /><br /></div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-04-01T12:00:00-05:00">Friday, April  1</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F04%2F01%2Fphylometa-from-r-udpate%2F&amp;title=Phylometa%20from%20R%20-%20UDPATE" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/03/25/bio-oracle/">Bio-ORACLE</a></h2>
  </header>
  
    <a href="http://www.oracle.ugent.be/index.html">Bio-ORACLE</a><div><br /></div><div>A new dataset available of geophysical, biotic and climate data. Should be fun to play with in R. </div>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-03-25T12:00:00-05:00">Friday, March 25</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F03%2F25%2Fbio-oracle%2F&amp;title=Bio-ORACLE" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
    <li>
      <article class="instapaper_body">
  <header>
    <h2 class="instapaper_title"><a href="/2011/03/23/basic-ggplot2-network-graphs-ver2/">basic ggplot2 network graphs - ver2</a></h2>
  </header>
  
    I posted last week a <a href="http://r-ecology.blogspot.com/2011/03/basic-ggplot2-network-graphs.html">simple function to plot networks using ggplot2 package</a>. Here is version 2. I still need to work on figuring out efficient vertex placement.<br /><br /><u>Changes in version 2:</u><br />-You have one of three options: use an igraph object, a matrix, or a dataframe (matrices will be converted to data frames within the function)<br /><br />-If you have data on food webs similar to that provided in the Takapoto dataset provided in the NetIndices package, you can set trophic = "TRUE", and gggraph will use the function TrophInd to assign trophic levels (the y axis value) to each vertex/node. You have to provide additional information along with this option such as what the imports and exports are, see NetIndices documentation.<br /><br />-I added some simple error checking.<br /><br />-if using method="df" and trophic="FALSE", x axis placement of vertices is now done using the function&nbsp;degreex (see inside the fxn), which sorts vertices according to their degree (so the least connected species are on the left of the graph; note that species with the same degree are not stacked on the y-axis because e.g., two vertices of degree=5 would get x=3 then x=4).<br /><br /><br /><br /><div style="overflow: auto;"><div class="geshifilter"><pre class="r geshifilter-R" style="font-family: monospace;"><span style="color: #666666; font-style: italic;">######### ggraph version 2</span><br /><a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/bipartite">bipartite</a><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/igraph">igraph</a><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/ggplot2">ggplot2</a><span style="color: #009900;">)</span> <br /><a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/NetIndices">NetIndices</a><span style="color: #009900;">)</span><br />&nbsp;<br /><span style="color: #666666; font-style: italic;"># gggraph, version 2</span><br /><span style="color: #666666; font-style: italic;"># g = an igraph graph object, a matrix, or data frame</span><br /><span style="color: #666666; font-style: italic;"># vplace = type of vertex placement assignment, one of rnorm, runif, etc.</span><br /><span style="color: #666666; font-style: italic;"># method = one of 'df' for data frame, "mat' for matrix or "igraph" for an igraph graph object</span><br /><span style="color: #666666; font-style: italic;"># trophic = TRUE or FALSE for using Netindices function TrophInd to determine trophic level (y value in graph)</span><br /><span style="color: #666666; font-style: italic;"># trophinames = columns in matrix or dataframe to use for calculating trophic level</span><br /><span style="color: #666666; font-style: italic;"># import = named or refereced by col# columns of matrix or dataframe to use for import argument of TrophInd</span><br /><span style="color: #666666; font-style: italic;"># export = named or refereced by col# columns of matrix or dataframe to use for export argument of TrophInd</span><br /><span style="color: #666666; font-style: italic;"># dead = named or refereced by col# columns of matrix or dataframe to use for dead argument of TrophInd</span><br />&nbsp;<br />gggraph &lt;- <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>g<span style="color: #339933;">,</span> vplace = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #339933;">,</span> method<span style="color: #339933;">,</span> trophic = <span style="color: blue;">"FALSE"</span><span style="color: #339933;">,</span> <br />     trophinames<span style="color: #339933;">,</span> import<span style="color: #339933;">,</span> export<span style="color: #009900;">)</span> <br /><span style="color: #009900;">{</span><br />degreex  &lt;- <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>x<span style="color: #009900;">)</span> <span style="color: #009900;">{</span><br />  degreecol &lt;- <a href="http://inside-r.org/r-doc/base/apply"><span style="color: #003399; font-weight: bold;">apply</span></a><span style="color: #009900;">(</span>x<span style="color: #339933;">,</span> <span style="color: #cc66cc;">2</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>y<span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>y<span style="color: #009900;">[</span>y&gt;<span style="color: #cc66cc;">0</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />  degreerow &lt;- <a href="http://inside-r.org/r-doc/base/apply"><span style="color: #003399; font-weight: bold;">apply</span></a><span style="color: #009900;">(</span>x<span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/function"><span style="color: #003399; font-weight: bold;">function</span></a><span style="color: #009900;">(</span>y<span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>y<span style="color: #009900;">[</span>y&gt;<span style="color: #cc66cc;">0</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />  degrees &lt;- <a href="http://inside-r.org/r-doc/base/sort"><span style="color: #003399; font-weight: bold;">sort</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span>degreecol<span style="color: #339933;">,</span> degreerow<span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />  <a href="http://inside-r.org/r-doc/stats/df"><span style="color: #003399; font-weight: bold;">df</span></a> &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>degrees<span style="color: #339933;">,</span> x = <a href="http://inside-r.org/r-doc/base/seq"><span style="color: #003399; font-weight: bold;">seq</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>degrees<span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />  <a href="http://inside-r.org/r-doc/stats/df"><span style="color: #003399; font-weight: bold;">df</span></a>$value &lt;- <a href="http://inside-r.org/r-doc/base/rownames"><span style="color: #003399; font-weight: bold;">rownames</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/stats/df"><span style="color: #003399; font-weight: bold;">df</span></a><span style="color: #009900;">)</span><br />  <a href="http://inside-r.org/r-doc/stats/df"><span style="color: #003399; font-weight: bold;">df</span></a><br /> <span style="color: #009900;">}</span><br />  <span style="color: #666666; font-style: italic;"># require igraph</span><br /> <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>!<a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/igraph">igraph</a><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><span style="color: blue;">"must first install 'igraph' package."</span><span style="color: #009900;">)</span> <br />   <span style="color: #666666; font-style: italic;"># require ggplot2</span><br /> <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>!<a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/ggplot2">ggplot2</a><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><span style="color: blue;">"must first install 'ggplot2' package."</span><span style="color: #009900;">)</span><br />&nbsp;<br /><span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>method==<span style="color: blue;">"df"</span><span style="color: #009900;">)</span><span style="color: #009900;">{</span><br />  <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/class"><span style="color: #003399; font-weight: bold;">class</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span>==<span style="color: blue;">"matrix"</span><span style="color: #009900;">)</span><span style="color: #009900;">{</span> g &lt;- <a href="http://inside-r.org/r-doc/base/as.data.frame"><span style="color: #003399; font-weight: bold;">as.data.frame</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span> <span style="color: #009900;">}</span><br />  <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/class"><span style="color: #003399; font-weight: bold;">class</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span>!=<span style="color: blue;">"data.frame"</span><span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><span style="color: blue;">"object must be of class 'data.frame.'"</span><span style="color: #009900;">)</span><br /><span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>trophic==<span style="color: blue;">"FALSE"</span><span style="color: #009900;">)</span><span style="color: #009900;">{</span><br /> <span style="color: #666666; font-style: italic;"># data preparation from adjacency matrix</span><br /> temp &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/expand.grid"><span style="color: #003399; font-weight: bold;">expand.grid</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/dimnames"><span style="color: #003399; font-weight: bold;">dimnames</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span>:<span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/as.vector"><span style="color: #003399; font-weight: bold;">as.vector</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/as.matrix"><span style="color: #003399; font-weight: bold;">as.matrix</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /> temp &lt;- temp<span style="color: #009900;">[</span><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span> &gt; <span style="color: #cc66cc;">0</span><span style="color: #009900;">)</span> &amp; !<a href="http://inside-r.org/r-doc/base/is.na"><span style="color: #003399; font-weight: bold;">is.na</span></a><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #009900;">]</span><br /> temp &lt;- temp<span style="color: #009900;">[</span><a href="http://inside-r.org/r-doc/base/sort.list"><span style="color: #003399; font-weight: bold;">sort.list</span></a><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #009900;">]</span><br /> g_df &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>rows = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> cols = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> freqint = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br />&nbsp;<br /> g_df$id &lt;- <span style="color: #cc66cc;">1</span>:<a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>id=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">4</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> rows=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> cols=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> freqint=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df_ &lt;- melt<span style="color: #009900;">(</span>g_df<span style="color: #339933;">,</span> id=<a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">4</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />&nbsp;<br /> xy_s &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>degreex<span style="color: #009900;">(</span>g<span style="color: #009900;">)</span><span style="color: #339933;">,</span> y = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df_$value<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <br /> g_df_2 &lt;- <a href="http://inside-r.org/r-doc/base/merge"><span style="color: #003399; font-weight: bold;">merge</span></a><span style="color: #009900;">(</span>g_df_<span style="color: #339933;">,</span> xy_s<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/by"><span style="color: #003399; font-weight: bold;">by</span></a> = <span style="color: blue;">"value"</span><span style="color: #009900;">)</span><br /> <span style="color: #009900;">}</span> <span style="color: black; font-weight: bold;">else</span> <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>trophic==<span style="color: blue;">"TRUE"</span><span style="color: #009900;">)</span> <span style="color: #009900;">{</span><br />   <span style="color: #666666; font-style: italic;"># require NetIndices</span><br />  <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>!<a href="http://inside-r.org/r-doc/base/require"><span style="color: #003399; font-weight: bold;">require</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/packages/cran/NetIndices">NetIndices</a><span style="color: #009900;">)</span><span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><span style="color: blue;">"must first install 'NetIndices' package."</span><span style="color: #009900;">)</span><br /> <span style="color: #666666; font-style: italic;"># data preparation from adjacency matrix</span><br /> temp &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/expand.grid"><span style="color: #003399; font-weight: bold;">expand.grid</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/dimnames"><span style="color: #003399; font-weight: bold;">dimnames</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">[</span>-trophinames<span style="color: #339933;">,</span> -trophinames<span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span>:<span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> <br />   <a href="http://inside-r.org/r-doc/base/as.vector"><span style="color: #003399; font-weight: bold;">as.vector</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/as.matrix"><span style="color: #003399; font-weight: bold;">as.matrix</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">[</span>-trophinames<span style="color: #339933;">,</span> -trophinames<span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /> temp &lt;- temp<span style="color: #009900;">[</span><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span> &gt; <span style="color: #cc66cc;">0</span><span style="color: #009900;">)</span> &amp; !<a href="http://inside-r.org/r-doc/base/is.na"><span style="color: #003399; font-weight: bold;">is.na</span></a><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #009900;">]</span><br /> temp &lt;- temp<span style="color: #009900;">[</span><a href="http://inside-r.org/r-doc/base/sort.list"><span style="color: #003399; font-weight: bold;">sort.list</span></a><span style="color: #009900;">(</span>temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <span style="color: #009900;">]</span><br /> g_df &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>rows = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> cols = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> freqint = temp<span style="color: #009900;">[</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br />&nbsp;<br /> g_df$id &lt;- <span style="color: #cc66cc;">1</span>:<a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>id=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">4</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> rows=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> <br />      cols=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">2</span><span style="color: #009900;">]</span><span style="color: #339933;">,</span> freqint=g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">3</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df_ &lt;- melt<span style="color: #009900;">(</span>g_df<span style="color: #339933;">,</span> id=<a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">4</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />&nbsp;<br /> xy_s &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>value = <a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df_$value<span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />  x = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df_$value<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />  y = TrophInd<span style="color: #009900;">(</span>g<span style="color: #339933;">,</span> Import=import<span style="color: #339933;">,</span> Export=export<span style="color: #009900;">)</span><span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df_2 &lt;- <a href="http://inside-r.org/r-doc/base/merge"><span style="color: #003399; font-weight: bold;">merge</span></a><span style="color: #009900;">(</span>g_df_<span style="color: #339933;">,</span> xy_s<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/by"><span style="color: #003399; font-weight: bold;">by</span></a> = <span style="color: blue;">"value"</span><span style="color: #009900;">)</span> <br /> <span style="color: #009900;">}</span><br /> <span style="color: #666666; font-style: italic;"># plotting</span><br /> p &lt;- <a href="http://inside-r.org/packages/cran/ggplot">ggplot</a><span style="color: #009900;">(</span>g_df_2<span style="color: #339933;">,</span> aes<span style="color: #009900;">(</span>x<span style="color: #339933;">,</span> y<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  geom_point<span style="color: #009900;">(</span>size = <span style="color: #cc66cc;">5</span><span style="color: #009900;">)</span> +<br />  geom_line<span style="color: #009900;">(</span>aes<span style="color: #009900;">(</span>size = freqint<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/grDevices/group"><span style="color: #003399; font-weight: bold;">group</span></a> = id<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  geom_text<span style="color: #009900;">(</span>size = <span style="color: #cc66cc;">3</span><span style="color: #339933;">,</span> hjust = <span style="color: #cc66cc;">1.5</span><span style="color: #339933;">,</span> aes<span style="color: #009900;">(</span>label = value<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  theme_bw<span style="color: #009900;">(</span><span style="color: #009900;">)</span> + <br />  opts<span style="color: #009900;">(</span>panel.grid.major=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />   panel.grid.minor=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />   axis.text.x=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.text.y=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.title.x=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.title.y=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.ticks=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   panel.border=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   legend.position=<span style="color: blue;">"none"</span><span style="color: #009900;">)</span><br />&nbsp;<br /> p <span style="color: #666666; font-style: italic;"># return graph</span><br /><span style="color: #009900;">}</span> <span style="color: black; font-weight: bold;">else</span> <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span>method==<span style="color: blue;">"igraph"</span><span style="color: #009900;">)</span> <span style="color: #009900;">{</span><br />  <span style="color: black; font-weight: bold;">if</span><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/class"><span style="color: #003399; font-weight: bold;">class</span></a><span style="color: #009900;">(</span>g<span style="color: #009900;">)</span>!=<span style="color: blue;">"igraph"</span><span style="color: #009900;">)</span> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><span style="color: blue;">"object must be of class 'igraph.'"</span><span style="color: #009900;">)</span><br /> <span style="color: #666666; font-style: italic;"># data preparation from igraph object</span><br /> g_ &lt;- get.edgelist<span style="color: #009900;">(</span>g<span style="color: #009900;">)</span><br /> g_df &lt;- <a href="http://inside-r.org/r-doc/base/as.data.frame"><span style="color: #003399; font-weight: bold;">as.data.frame</span></a><span style="color: #009900;">(</span>g_<span style="color: #009900;">)</span><br /> g_df$id &lt;- <span style="color: #cc66cc;">1</span>:<a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span>g_df<span style="color: #009900;">[</span><span style="color: #339933;">,</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /> g_df &lt;- melt<span style="color: #009900;">(</span>g_df<span style="color: #339933;">,</span> id=<span style="color: #cc66cc;">3</span><span style="color: #009900;">)</span><br /> xy_s &lt;- <a href="http://inside-r.org/r-doc/base/data.frame"><span style="color: #003399; font-weight: bold;">data.frame</span></a><span style="color: #009900;">(</span>value = <a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df$value<span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />  x = vplace<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df$value<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />  y = vplace<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/length"><span style="color: #003399; font-weight: bold;">length</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/unique"><span style="color: #003399; font-weight: bold;">unique</span></a><span style="color: #009900;">(</span>g_df$value<span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /> g_df2 &lt;- <a href="http://inside-r.org/r-doc/base/merge"><span style="color: #003399; font-weight: bold;">merge</span></a><span style="color: #009900;">(</span>g_df<span style="color: #339933;">,</span> xy_s<span style="color: #339933;">,</span> <a href="http://inside-r.org/r-doc/base/by"><span style="color: #003399; font-weight: bold;">by</span></a> = <span style="color: blue;">"value"</span><span style="color: #009900;">)</span> <br />&nbsp;<br /> <span style="color: #666666; font-style: italic;"># plotting</span><br /> p &lt;- <a href="http://inside-r.org/packages/cran/ggplot">ggplot</a><span style="color: #009900;">(</span>g_df2<span style="color: #339933;">,</span> aes<span style="color: #009900;">(</span>x<span style="color: #339933;">,</span> y<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  geom_point<span style="color: #009900;">(</span>size = <span style="color: #cc66cc;">2</span><span style="color: #009900;">)</span> +<br />  geom_line<span style="color: #009900;">(</span>size = <span style="color: #cc66cc;">0.3</span><span style="color: #339933;">,</span> aes<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/grDevices/group"><span style="color: #003399; font-weight: bold;">group</span></a> = id<span style="color: #339933;">,</span> linetype = id<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  geom_text<span style="color: #009900;">(</span>size = <span style="color: #cc66cc;">3</span><span style="color: #339933;">,</span> hjust = <span style="color: #cc66cc;">1.5</span><span style="color: #339933;">,</span> aes<span style="color: #009900;">(</span>label = value<span style="color: #009900;">)</span><span style="color: #009900;">)</span> +<br />  theme_bw<span style="color: #009900;">(</span><span style="color: #009900;">)</span> + <br />  opts<span style="color: #009900;">(</span>panel.grid.major=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />   panel.grid.minor=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />   axis.text.x=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.text.y=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.title.x=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.title.y=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   axis.ticks=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   panel.border=theme_blank<span style="color: #009900;">(</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span><br />   legend.position=<span style="color: blue;">"none"</span><span style="color: #009900;">)</span><br />&nbsp;<br /> p <span style="color: #666666; font-style: italic;"># return graph</span><br /><span style="color: #009900;">}</span> <span style="color: black; font-weight: bold;">else</span> <br /> <a href="http://inside-r.org/r-doc/base/stop"><span style="color: #003399; font-weight: bold;">stop</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/base/paste"><span style="color: #003399; font-weight: bold;">paste</span></a><span style="color: #009900;">(</span><span style="color: blue;">"do not recognize method = <span style="color: #000099; font-weight: bold;">\"</span>"</span><span style="color: #339933;">,</span>method<span style="color: #339933;">,</span><span style="color: blue;">"<span style="color: #000099; font-weight: bold;">\"</span>; <br />  methods are <span style="color: #000099; font-weight: bold;">\"</span>df<span style="color: #000099; font-weight: bold;">\"</span> and <span style="color: #000099; font-weight: bold;">\"</span>igraph<span style="color: #000099; font-weight: bold;">\"</span>"</span><span style="color: #339933;">,</span>sep=<span style="color: blue;">""</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /><span style="color: #009900;">}</span></pre></div></div><br /><span style="color: #666666; font-style: italic;">############### Eg</span><br /><a href="http://inside-r.org/r-doc/utils/data"><span style="color: #003399; font-weight: bold;">data</span></a><span style="color: #009900;">(</span>Takapoto<span style="color: #009900;">)</span><br />gggraph<span style="color: #009900;">(</span>Takapoto<span style="color: #339933;">,</span> vplace = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #339933;">,</span> method = <span style="color: blue;">"df"</span><span style="color: #339933;">,</span> trophic = <span style="color: blue;">"TRUE"</span><span style="color: #339933;">,</span> trophinames = <a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">8</span>:<span style="color: #cc66cc;">10</span><span style="color: #009900;">)</span><span style="color: #339933;">,</span> <br />import = <span style="color: blue;">"CO2"</span><span style="color: #339933;">,</span> export = <a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: blue;">"CO2"</span><span style="color: #339933;">,</span> <span style="color: blue;">"Sedimentation"</span><span style="color: #339933;">,</span> <span style="color: blue;">"Grazing"</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /><div class="separator" style="clear: both; text-align: center;"><br /></div><br /><br /><div class="separator" style="clear: both; text-align: center;"><a href="https://lh3.googleusercontent.com/-LAeWjCzUxzU/TYpCkoo2BwI/AAAAAAAAEbQ/5v5696gOia8/s1600/gggraph_trophic.jpeg" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="400" src="https://lh3.googleusercontent.com/-LAeWjCzUxzU/TYpCkoo2BwI/AAAAAAAAEbQ/5v5696gOia8/s400/gggraph_trophic.jpeg" width="400" /></a></div><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;">&nbsp;<br /></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;">plants &lt;- <a href="http://inside-r.org/r-doc/base/round"><span style="color: #003399; font-weight: bold;">round</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/stats/rlnorm"><span style="color: #003399; font-weight: bold;">rlnorm</span></a><span style="color: #009900;">(</span>n=<span style="color: #cc66cc;">5</span><span style="color: #339933;">,</span> meanlog=<span style="color: #cc66cc;">2</span><span style="color: #339933;">,</span> sdlog=<span style="color: #cc66cc;">1</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/cluster/animals"><span style="color: #003399; font-weight: bold;">animals</span></a> &lt;- <a href="http://inside-r.org/r-doc/base/round"><span style="color: #003399; font-weight: bold;">round</span></a><span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/stats/rlnorm"><span style="color: #003399; font-weight: bold;">rlnorm</span></a><span style="color: #009900;">(</span>n=<span style="color: #cc66cc;">5</span><span style="color: #339933;">,</span> meanlog=<span style="color: #cc66cc;">2</span><span style="color: #339933;">,</span> sdlog=<span style="color: #cc66cc;">1</span><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />plants &lt;- plants*<span style="color: #009900;">(</span><span style="color: #cc66cc;">100</span>/sum<span style="color: #009900;">(</span>plants<span style="color: #009900;">)</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/cluster/animals"><span style="color: #003399; font-weight: bold;">animals</span></a> &lt;- <a href="http://inside-r.org/r-doc/cluster/animals"><span style="color: #003399; font-weight: bold;">animals</span></a>*<span style="color: #009900;">(</span><span style="color: #cc66cc;">100</span>/sum<span style="color: #009900;">(</span><a href="http://inside-r.org/r-doc/cluster/animals"><span style="color: #003399; font-weight: bold;">animals</span></a><span style="color: #009900;">)</span><span style="color: #009900;">)</span><br />z &lt;- <a href="http://inside-r.org/r-doc/stats/r2dtable"><span style="color: #003399; font-weight: bold;">r2dtable</span></a><span style="color: #009900;">(</span><span style="color: #cc66cc;">1</span><span style="color: #339933;">,</span><a href="http://inside-r.org/r-doc/cluster/animals"><span style="color: #003399; font-weight: bold;">animals</span></a><span style="color: #339933;">,</span>plants<span style="color: #009900;">)</span>  <span style="color: #666666; font-style: italic;"># if you get errors on this step just rerun again until no error</span><br />z &lt;- <a href="http://inside-r.org/r-doc/base/as.data.frame"><span style="color: #003399; font-weight: bold;">as.data.frame</span></a><span style="color: #009900;">(</span>z<span style="color: #009900;">[</span><span style="color: #009900;">[</span><span style="color: #cc66cc;">1</span><span style="color: #009900;">]</span><span style="color: #009900;">]</span><span style="color: #009900;">)</span><br /><a href="http://inside-r.org/r-doc/base/rownames"><span style="color: #003399; font-weight: bold;">rownames</span></a><span style="color: #009900;">(</span>z<span style="color: #009900;">)</span> &lt;- <a href="http://inside-r.org/r-doc/base/c"><span style="color: #003399; font-weight: bold;">c</span></a><span style="color: #009900;">(</span><span style="color: blue;">"a"</span><span style="color: #339933;">,</span><span style="color: blue;">"b"</span><span style="color: #339933;">,</span><span style="color: blue;">"c"</span><span style="color: #339933;">,</span><span style="color: blue;">"d"</span><span style="color: #339933;">,</span><span style="color: blue;">"e"</span><span style="color: #009900;">)</span><br />gggraph<span style="color: #009900;">(</span>z<span style="color: #339933;">,</span> vplace = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #339933;">,</span> method = <span style="color: blue;">"df"</span><span style="color: #339933;">,</span> trophic = <span style="color: blue;">"FALSE"</span><span style="color: #009900;">)</span><br /></pre><div class="separator" style="clear: both; text-align: center;"><a href="https://lh6.googleusercontent.com/-FSpM2FmvOeE/TYpCkQySuAI/AAAAAAAAEbM/TSSrg2HkoBU/s1600/gggraph_matrixdf.jpeg" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="400" src="https://lh6.googleusercontent.com/-FSpM2FmvOeE/TYpCkQySuAI/AAAAAAAAEbM/TSSrg2HkoBU/s400/gggraph_matrixdf.jpeg" width="400" /></a></div><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;">&nbsp;<br /></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;"></pre><pre class="r geshifilter-R" style="font-family: monospace;">g &lt;- grg.game<span style="color: #009900;">(</span><span style="color: #cc66cc;">20</span><span style="color: #339933;">,</span> <span style="color: #cc66cc;">0.45</span><span style="color: #339933;">,</span> torus=<span style="color: black; font-weight: bold;">FALSE</span><span style="color: #009900;">)</span><br />gggraph<span style="color: #009900;">(</span>g<span style="color: #339933;">,</span> vplace = <a href="http://inside-r.org/r-doc/stats/rnorm"><span style="color: #003399; font-weight: bold;">rnorm</span></a><span style="color: #339933;">,</span> method = <span style="color: blue;">"igraph"</span><span style="color: #339933;">,</span> trophic = <span style="color: blue;">"FALSE"</span><span style="color: #009900;">)</span></pre><pre class="r geshifilter-R" style="font-family: monospace;"><span style="color: #009900;"><br /></span></pre><div class="separator" style="clear: both; text-align: center;"><a href="https://lh5.googleusercontent.com/-VSD2Xdt7Ukk/TYpCj8hbNsI/AAAAAAAAEbI/-YbX6s0I2QY/s1600/gggraph_igraph.jpeg" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="400" src="https://lh5.googleusercontent.com/-VSD2Xdt7Ukk/TYpCj8hbNsI/AAAAAAAAEbI/-YbX6s0I2QY/s400/gggraph_igraph.jpeg" width="400" /></a></div><pre class="r geshifilter-R" style="font-family: monospace;"><span style="color: #009900;"><span class="Apple-style-span" style="color: black; font-family: Times; white-space: normal;"></span></span></pre><pre class="r geshifilter-R" style="font-family: monospace;"><span style="color: #009900;"><br /></span></pre><a href="http://www.inside-r.org/pretty-r" title="Created by Pretty R at inside-R.org">Created by Pretty R at inside-R.org</a>

  
  <footer class="group">
    <p>
      
      
      
        
      
      <time pubdate datetime="2011-03-23T12:00:00-05:00">Wednesday, March 23</time>
      
      <a class="icon" href="http://www.instapaper.com/hello2?u=http%3A%2F%2Fschamberlain.github.com%2Frecology%2F%2F2011%2F03%2F23%2Fbasic-ggplot2-network-graphs-ver2%2F&amp;title=basic%20ggplot2%20network%20graphs%20-%20ver2" rel="alternate" title="Save to Instapaper">Read Later</a>
      
    </p>
  </footer>
</article>


    </li>
    
  </ol>

  
  <nav id="pagination">
    <ol>
      
        
      
      
      <li>
        <a href="/pagelayoutdefaulturl/page7/index.htmlcontent<section id="posts">
  <ol>
    {% for post in paginator.posts %}
    <li>
      {% include post.html %}
    </li>
    {% endfor %}
  </ol>

  {% if paginator.total_pages > 1 %}
  <nav id="pagination">
    <ol>
      {% if paginator.previous_page > 1 %}
        {% capture page_url %}page{{ page }}{% endcapture%}
      {% endif %}
      {% if paginator.previous_page %}
      <li>
        <a href="/{{ page_url }}" title="Go to previous page">
          &laquo;
        </a>
      </li>
      {% endif %}
      <li>
      {% for page in (1..paginator.total_pages) %}
      <li>
        {% if page > 1 %}
          {% capture page_url %}page{{ page }}{% endcapture%}
        {% endif %}
        {% if page == paginator.page %}
          {{ page }}
        {% else %}
          <a href="/{{ page_url }}">{{ page }}</a>
        {% endif %}
      </li>
      {% endfor %}
      {% if paginator.next_page %}
      <li>
        <a href="/page{{ paginator.next_page }}" title="Go to next page">
          &raquo;
        </a>
      </li>
      {% endif %}
    </ol>
  </nav>
  {% endif %}
</section>
" title="Go to previous page">
          &laquo;
        </a>
      </li>
      
      <li>
      
      <li>
        
        
          <a href="/pagelayoutdefaulturl/page7/index.htmlcontent<section id="posts">
  <ol>
    {% for post in paginator.posts %}
    <li>
      {% include post.html %}
    </li>
    {% endfor %}
  </ol>

  {% if paginator.total_pages > 1 %}
  <nav id="pagination">
    <ol>
      {% if paginator.previous_page > 1 %}
        {% capture page_url %}page{{ page }}{% endcapture%}
      {% endif %}
      {% if paginator.previous_page %}
      <li>
        <a href="/{{ page_url }}" title="Go to previous page">
          &laquo;
        </a>
      </li>
      {% endif %}
      <li>
      {% for page in (1..paginator.total_pages) %}
      <li>
        {% if page > 1 %}
          {% capture page_url %}page{{ page }}{% endcapture%}
        {% endif %}
        {% if page == paginator.page %}
          {{ page }}
        {% else %}
          <a href="/{{ page_url }}">{{ page }}</a>
        {% endif %}
      </li>
      {% endfor %}
      {% if paginator.next_page %}
      <li>
        <a href="/page{{ paginator.next_page }}" title="Go to next page">
          &raquo;
        </a>
      </li>
      {% endif %}
    </ol>
  </nav>
  {% endif %}
</section>
">1</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page2">2</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page3">3</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page4">4</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page5">5</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page6">6</a>
        
      </li>
      
      <li>
        
          
        
        
          7
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page8">8</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page9">9</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page10">10</a>
        
      </li>
      
      <li>
        
          
        
        
          <a href="/page11">11</a>
        
      </li>
      
      
      <li>
        <a href="/page8" title="Go to next page">
          &raquo;
        </a>
      </li>
      
    </ol>
  </nav>
  
</section>

        </section>

        <section id="sidebar">

          <section id="meta-col1" class="meta group">
            <aside id="about">
              <header>
                <h3>About</h3>
              </header>
              <p>
                Recology is a blog that explores using R for ecology, evolution, and open science. <a href="/about">See more.</a>
              </p>
            </aside>

            <aside>
              <header>
                <h3>Stay up to date</h3>
              </header>
              <nav>
                <ul>
                  <li class="icon rss"><a href="http://feeds.feedburner.com/XX">RSS</a></li>
                  <li class="icon twitter"><a href="http://twitter.com/recology_">Twitter</a></li>
                  <li class="icon"><a href="/subscribe">More</a></li>
                </ul>
              </nav>
            </aside>

          </section>

          <section id="meta-col2" class="meta group">

            <aside id="bio">
              <header>
                <h3>The Author</h3>
              </header>
              <p>
              My name is Scott Chamberlain. I am a researcher in ecology and evolution.
              </p>
            </aside>

            <aside>
              <nav id="extra">
                <ul>
                  <li><a href="/colophon">Colophon</a></li>
                  <li><a href="/contact">Contact</a></li>
                </ul>
              </nav>
            </aside>

          </section>

        </section>

      </section>

      <section id="footer" class="group">
        <footer class="group">
          <nav id="site">
            <ul>
              <li><a href="/">Home</a></li>
              <li><a href="/about">About</a></li>
              <li><a href="/contact">Contact</a></li>
              <li><a href="/subscribe">Subscribe</a></li>
              <li><a href="/colophon">Colophon</a></li>
            </ul>
          </nav>
          <p class="copyright">
            &copy;2011 &ndash; Scott Chamberlain
          </p>
        </footer>
      </section>

    </section>

    <script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-23094691-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();

    </script>

  </body>
</html>
