<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>Phylogenetics | Recology</title>
<meta name="keywords" content="">
<meta name="description" content="">
<meta name="author" content="">
<link rel="canonical" href="http://localhost:1313/tags/phylogenetics/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.a4ec0cdfac731fabf363c0b5d5ab206254bbf9b6aaf985d695c6bb534196df16.css" integrity="sha256-pOwM36xzH6vzY8C11asgYlS7&#43;baq&#43;YXWlca7U0GW3xY=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css">
<link rel="alternate" type="application/rss+xml" href="http://localhost:1313/tags/phylogenetics/index.xml">
<link rel="alternate" hreflang="en" href="http://localhost:1313/tags/phylogenetics/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="list" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Recology (Alt + H)">Recology</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/" title="Home">
                    <span>Home</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/links" title="Links">
                    <span>Links</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/posts/index.xml" title="RSS">
                    <span>RSS</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main"> 
<header class="page-header">
  <h1>
    Phylogenetics
  </h1>
</header>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Function for phylogeny resolution
    </h2>
  </header>
  <div class="entry-content">
    <p>UPDATE: Yeah, so the treeresstats function had a problem in one of the calculations. I fixed that and added some more calulcations to the function.
I couldn’t find any functions to calculate number of polytomies, and related metrics.
Here’s a simple function that gives four metrics on a phylo tree object:
# calculate tree resolution stats treeresstats &lt;- function(x) { require(phangorn) # load the phangorn package todo &lt;- ( 1&#43;Ntip(x)) : (Ntip(x) &#43; Nnode(x) ) trsize_tips &lt;- Ntip(x) trsize_nodes &lt;- Nnode(x) polytomyvec &lt;- sapply(todo, function(y) length(Children(x, y))) numpolys &lt;- length(polytomyvec[polytomyvec &gt; 2]) numpolysbytrsize_tips &lt;- numpolys/trsize_tips numpolysbytrsize_nodes &lt;- numpolys/trsize_nodes proptipsdescpoly &lt;- sum(polytomyvec[polytomyvec &gt; 2])/trsize_tips propnodesdich &lt;- length(polytomyvec[polytomyvec == 2])/trsize_nodes list(trsize_tips = trsize_tips, trsize_nodes = trsize_nodes, numpolys = numpolys, numpolysbytrsize_tips = numpolysbytrsize_tips, numpolysbytrsize_nodes = numpolysbytrsize_nodes, proptipsdescpoly = proptipsdescpoly, propnodesdich = propnodesdich) } # Single tree example tree &lt;- read.tree(text=&#34;((((((artemisia_species:44,lactuca_species:44,senecio_species:44)6:46,campanula_species:90)5:17.75,((asclepias_species:71,galium_species:71)8:18.375,plantago_species:89.375)7:18.375)4:17.75,((cerastium_species:41.833332,silene_species:41.833332)10:41.833332,chenopodium_species:83.666664)9:41.833336)3:17.75,((geum_species:47,potentilla_species:47)12:48.125,lepidium_species:95.125)11:48.125)2:17.75,(bromus_species:12,elymus_species:12)13:149)1;&#34;) dat &lt;- treeresstats(tree) dat # Many trees example maketrees &lt;- function(numtrees) { require(ape); require(plyr) trees &lt;- rmtree(numtrees, 20) llply(trees, di2multi, tol = 0.5) } trees &lt;- maketrees(30) dat &lt;- ldply(trees, function(x) data.frame(treeresstats(x))) dat Here’s output from the gist above:
...</p>
  </div>
  <footer class="entry-footer"><span title='2012-01-13 00:00:00 +0000 UTC'>January 13, 2012</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Function for phylogeny resolution" href="http://localhost:1313/2012/01/phylogeny-resolution/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">My talk on doing phylogenetics in R
    </h2>
  </header>
  <div class="entry-content">
    <p>I gave a talk today on doing very basic phylogenetics in R, including getting sequence data, aligning sequence data, plotting trees, doing trait evolution stuff, etc.
Please comment if you have code for doing bayesian phylogenetic inference in R. I know phyloch has function mrbayes, but can’t get it to work…
Phylogenetics in R View more presentations from schamber </p>
  </div>
  <footer class="entry-footer"><span title='2011-11-18 13:15:00.001 -0600 -0600'>November 18, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to My talk on doing phylogenetics in R" href="http://localhost:1313/2011/11/my-talk-on-doing-phylogenetics-in-r/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Jonathan Eisen on the Fourth Domain and Open Science
    </h2>
  </header>
  <div class="entry-content">
    <p>
Stalking the Fourth Domain with Jonathan Eisen, Ph D from mendelspod on Vimeo.</p>
  </div>
  <footer class="entry-footer"><span title='2011-09-06 18:53:00 -0500 -0500'>September 6, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Jonathan Eisen on the Fourth Domain and Open Science" href="http://localhost:1313/2011/09/jonathan-eisen-on-fourth-domain-and/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Tenure track position in systematics at the University of Vermont
    </h2>
  </header>
  <div class="entry-content">
    <p>There is an awesome position opening up for an assistant professor in systematics at the University of Vermont. Below is the announcement, and see the original post at the Distributed Ecology blog. Why is this related to R? One can do a lot of systematics work in R, including retrieving scientific collections data through an upcoming package handshaking with VertNet (part of the rOpenSci project), managing large data sets, retrieval of GenBank data through the ape package (see fxn read.genbank), phylogenetic reconstruction and analysis, and more. So I am sure a systematist with R ninja skills will surely have a head up on the rest of the field.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-08-22 15:24:00 -0500 -0500'>August 22, 2011</span>&nbsp;·&nbsp;3 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Tenure track position in systematics at the University of Vermont" href="http://localhost:1313/2011/08/tenure-track-position-in-systematics-at/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">iEvoBio 2011 Synopsis
    </h2>
  </header>
  <div class="entry-content">
    <p>We just wrapped up the 2011 iEvoBio meeting. It was awesome! If you didn’t go this year or last year, definitely think about going next year.
Here is a list of the cool projects that were discussed at the meeting (apologies if I left some out):
Vistrails: workflow tool, awesome project by Claudio Silva Commplish: purpose is to use via API’s, not with the web UI Phylopic: a database of life-form silouhettes, including an API for remote access, sweet! Gloome MappingLife: awesome geographic/etc data visualization interace on the web SuiteSMA: visualizating multiple alignments treeBASE: R interface to treebase, by Carl Boettiger VertNet: database for vertebrate natural history collections RevBayes: revamp of MrBayes, with GUI, etc. Phenoscape Knowledge Base Peter Midford lightning talk: talked about matching taxonomic and genetic data BiSciCol: biological science collections tracker Ontogrator TNRS: taxonomic name resolution service Barcode of Life data systems, and remote access Moorea Biocode Project Microbial LTER’s data BirdVis: interactive bird data visualization (Claudio Silva in collaboration with Cornell Lab of Ornithology) Crowdlabs: I think the site is down right now, another project by Claudio Silva Phycas: Bayesian phylogenetics, can you just call this from R? RIP MrBayes!!!! replaced by RevBayes (see 9 above) Slides of presentations will be at Slideshare (not all presentations up yet) A birds of a feather group I was involved in proposed an idea (TOL-o-matic) like Phylomatic, but of broader scope, for easy access and submission of trees, and perhaps even social (think just pushing a ‘SHARE’ button within PAUP, RevBayes, or other phylogenetics software)! Synopses of Birds of a Feather discussion groups: http://piratepad.net/iEvoBio11-BoF-reportouts </p>
  </div>
  <footer class="entry-footer"><span title='2011-06-22 21:38:00.001 -0500 -0500'>June 22, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to iEvoBio 2011 Synopsis" href="http://localhost:1313/2011/06/ievobio-2011-synopsis/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">searching ITIS and fetching Phylomatic trees
    </h2>
  </header>
  <div class="entry-content">
    <p>I am writing a set of functions to search ITIS for taxonomic information (more databases to come) and functions to fetch plant phylogenetic trees from Phylomatic. Code at github.
Also, see the examples in the demos folder on the Github site above.
</p>
  </div>
  <footer class="entry-footer"><span title='2011-06-03 17:35:00.001 -0500 -0500'>June 3, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to searching ITIS and fetching Phylomatic trees" href="http://localhost:1313/2011/06/searching-itis-and-fetching-phylomatic-trees/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">phylogenetic signal simulations
    </h2>
  </header>
  <div class="entry-content">
    <p>I did a little simulation to examine how K and lambda vary in response to tree size (and how they compare to each other on the same simulated trees). I use Liam Revell’s functions fastBM to generate traits, and phylosig to measure phylogenetic signal.
Two observations:
First, it seems that lambda is more sensitive than K to tree size, but then lambda levels out at about 40 species, whereas K continues to vary around a mean of 1. Second, K is more variable than lambda at all levels of tree size (compare standard error bars). Does this make sense to those smart folks out there?
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-18 08:13:00 -0500 -0500'>May 18, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to phylogenetic signal simulations" href="http://localhost:1313/2011/05/phylogenetic-signal-simulations/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">A simple function for plotting phylogenies in ggplot2
    </h2>
  </header>
  <div class="entry-content">
    <p>UPDATE: Greg jordan has a much more elegant way of plotting trees with ggplot2. See his links in the comments below.
I wrote a simple function for plotting a phylogeny in ggplot2. However, it only handles a 3 species tree right now, as I haven’t figured out how to generalize the approach to N species. It’s at https://gist.github.com/977207
Any ideas on how to improve this?
</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-17 14:37:00.001 -0500 -0500'>May 17, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to A simple function for plotting phylogenies in ggplot2" href="http://localhost:1313/2011/05/simple-function-for-plotting/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Comparison of functions for comparative phylogenetics
    </h2>
  </header>
  <div class="entry-content">
    <p>With all the packages (and beta stage groups of functions) for comparative phylogenetics in R (tested here: picante, geiger, ape, motmot, Liam Revell’s functions), I was simply interested in which functions to use in cases where multiple functions exist to do the same thing. I only show default settings, so perhaps these functions would differ under different parameter settings. [I am using a Mac 2.4 GHz i5, 4GB RAM]
Get motmot here: https://r-forge.r-project.org/R/?group_id=782
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-11 10:08:00 -0500 -0500'>May 11, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Comparison of functions for comparative phylogenetics" href="http://localhost:1313/2011/05/comparison-of-functions-for-comparative-phylogenetics/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Treebase trees from R
    </h2>
  </header>
  <div class="entry-content">
    <p>UPDATE: See Carl Boettiger’s functions/package at Github for searching Treebase here.
Treebase is a great resource for phylogenetic trees, and has a nice interface for searching for certain types of trees. However, if you want to simply download a lot of trees for analyses (like that in Davies et al.), then you want to be able to access trees in bulk (I believe Treebase folks are working on an API though). I wrote some simple code for extracting trees from Treebase.org.It reads an xml file of (in this case consensus) URL’s for each tree, parses the xml, makes a vector of URL’s, reads the nexus files with error checking, remove trees that gave errors, then a simple plot looking at metrics of the trees.
...</p>
  </div>
  <footer class="entry-footer"><span title='2011-05-03 10:00:00.001 -0500 -0500'>May 3, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Treebase trees from R" href="http://localhost:1313/2011/05/treebase-trees-from-r/"></a>
</article>
<footer class="page-footer">
  <nav class="pagination">
    <a class="next" href="http://localhost:1313/tags/phylogenetics/page/2/">Next&nbsp;&nbsp;»
    </a>
  </nav>
</footer>
    </main>
    
<footer class="footer">
        <span>&copy; 2025 <a href="http://localhost:1313/">Recology</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
