<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>R | Recology</title>
<meta name="keywords" content="">
<meta name="description" content="">
<meta name="author" content="">
<link rel="canonical" href="http://localhost:1313/tags/r/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.f4cda22486791d430ef692108ba33a982c3c552eca57e99456bb55aa4bd22568.css" integrity="sha256-9M2iJIZ5HUMO9pIQi6M6mCw8VS7KV&#43;mUVrtVqkvSJWg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" type="application/rss+xml" href="http://localhost:1313/tags/r/index.xml">
<link rel="alternate" hreflang="en" href="http://localhost:1313/tags/r/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="list" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Recology (Alt + H)">Recology</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/" title="Home">
                    <span>Home</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/index.xml" title="RSS">
                    <span>RSS</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main"> 
<header class="page-header">
  <h1>
    R
  </h1>
</header>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Running Phylip&#39;s contrast application for trait pairs from R
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is some code to run Phylip’s contrast application from R and get the output within R to easily manipulate yourself. Importantly, the code is written specifically for trait pairs only as the regular expression work in the code specifically grabs data from contast results when only two traits are input. You could easily change the code to do N traits. Note that the p-value calculated for the chi-square statistic is not output from contrast, but is calculated within the function ‘PhylipWithinSpContr’....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-26 07:40:00.004 -0500 -0500'>April 26, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Running Phylip&#39;s contrast application for trait pairs from R" href="http://localhost:1313/2011/04/running-phylip-s-contrast-application-for-trait-pairs-from-r/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R: Randomization via Tip Shuffle
    </h2>
  </header>
  <div class="entry-content">
    <p>—UPDATE: I am now using code formatting from gist.github, so I replaced the old prettyR code (sorry guys). The github way is much easier and prettier. I hope readers like the change.
I wrote earlier about some code I wrote for running Phylometa (software to do phylogenetic meta-analysis) from R.
I have been concerned about what exactly is the right penalty for including phylogeny in a meta-analysis. E.g.: AIC is calculated from Q in Phylometa, and Q increases with tree size....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-16 13:44:00.004 -0500 -0500'>April 16, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R: Randomization via Tip Shuffle" href="http://localhost:1313/2011/04/phylometa-from-r-randomization-via-tip-shuffle/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Adjust branch lengths with node ages: comparison of two methods
    </h2>
  </header>
  <div class="entry-content">
    <p>Here is an approach for comparing two methods of adjusting branch lengths on trees: bladj in the program Phylocom and a fxn written by Gene Hunt at the Smithsonian.
Get the code and example files (tree and node ages) at https://gist.github.com/938313
Get phylocom at http://www.phylodiversity.net/phylocom/
Gene Hunt’s method has many options you can mess with, including setting tip ages (not available in bladj), setting node ages, and minimum branch length imposed....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-10 13:02:00.003 -0500 -0500'>April 10, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Adjust branch lengths with node ages: comparison of two methods" href="http://localhost:1313/2011/04/adjust-branch-lengths-with-node-ages-comparison-of-two-methods/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylometa from R - UDPATE
    </h2>
  </header>
  <div class="entry-content">
    <p>A while back I posted some messy code to run Phylometa from R, especially useful for processing the output data from Phylometa which is not easily done. The code is still quite messy, but it should work now. I have run the code with tens of different data sets and phylogenies so it should work.
I fixed errors when parentheses came up against numbers in the output, and other things. You can use the code for up to 4 levels of your grouping variable....</p>
  </div>
  <footer class="entry-footer"><span title='2011-04-01 16:18:00.003 -0500 -0500'>April 1, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylometa from R - UDPATE" href="http://localhost:1313/2011/04/phylometa-from-r-udpate/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">basic ggplot2 network graphs ver2
    </h2>
  </header>
  <div class="entry-content">
    <p>I posted last week a simple function to plot networks using ggplot2 package. Here is version 2. I still need to work on figuring out efficient vertex placement.
Changes in version 2:
You have one of three options: use an igraph object, a matrix, or a dataframe (matrices will be converted to data frames within the function) If you have data on food webs similar to that provided in the Takapoto dataset provided in the NetIndices package, you can set trophic = “TRUE”, and gggraph will use the function TrophInd to assign trophic levels (the y axis value) to each vertex/node....</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-23 14:09:00 -0500 -0500'>March 23, 2011</span>&nbsp;·&nbsp;5 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to basic ggplot2 network graphs ver2" href="http://localhost:1313/2011/03/basic-ggplot2-network-graphs-ver2/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">basic ggplot2 network graphs
    </h2>
  </header>
  <div class="entry-content">
    <p>I have been looking around on the web and have not found anything yet related to using ggplot2 for making graphs/networks. I put together a few functions to make very simple graphs. The bipartite function especially is not ideal, as of course we only want to allow connections between unlike nodes, not all nodes. These functions do not, obviously, take full advantage of the power of ggplot2, but it’s a start....</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-17 08:35:00 -0500 -0500'>March 17, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to basic ggplot2 network graphs" href="http://localhost:1313/2011/03/basic-ggplot2-network-graphs/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Five ways to visualize your pairwise comparisons
    </h2>
  </header>
  <div class="entry-content">
    <p>UPDATE: At the bottom are two additional methods, and some additions (underlined) are added to the original 5 methods. Thanks for all the feedback…
-Also, another post here about ordered-categorical data-Also #2, a method combining splom and hexbin packages here, for larger datasets
In data analysis it is often nice to look at all pairwise combinations of continuous variables in scatterplots. Up until recently, I have used the function splom in the package lattice, but ggplot2 has superior aesthetics, I think anyway....</p>
  </div>
  <footer class="entry-footer"><span title='2011-03-05 12:49:00.006 -0600 -0600'>March 5, 2011</span>&nbsp;·&nbsp;3 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Five ways to visualize your pairwise comparisons" href="http://localhost:1313/2011/03/for-all-your-pairwise-comparison-needs/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phenotypic selection analysis in R
    </h2>
  </header>
  <div class="entry-content">
    <p>I have up to recently always done my phenotypic selection analyses in SAS. I finally got some code I think works to do everything SAS would do. Feedback much appreciated!
########################Selection analyses############################# install.packages(c(&#34;car&#34;,&#34;reshape&#34;,&#34;ggplot2&#34;)) require(car) require(reshape) require(ggplot2) # Create data set dat &lt;- data.frame(plant = seq(1,100,1), trait1 = rep(c(0.1,0.15,0.2,0.21,0.25,0.3,0.5,0.6,0.8,0.9,1,3,4,10,11,12,13,14,15,16), each = 5), trait2 = runif(100), fitness = rep(c(1,5,10,20,50), each = 20)) # Make relative fitness column dat_ &lt;- cbind(dat, dat$fitness/mean(dat$fitness)) names(dat_)[5] &lt;- &#34;relfitness&#34; # Standardize traits dat_ &lt;- cbind(dat_[,-c(2:3)], rescaler(dat_[,c(2:3)],&#34;sd&#34;)) ####Selection differentials and correlations among traits, cor....</p>
  </div>
  <footer class="entry-footer"><span title='2011-02-24 09:43:00.001 -0600 -0600'>February 24, 2011</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phenotypic selection analysis in R" href="http://localhost:1313/2011/02/phenotypic-selection-analysis-in-r/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Phylogenetic analysis with the phangorn package: an example
    </h2>
  </header>
  <div class="entry-content">
    <p>The phangorn package is a relatively new package in R for the analysis and comparison of phylogenies. See here for the Bioinformatics paper and here for the package. Here is an example of using phangorn from getting sequences to making phylogenies and visualizing them:Getting sequences from GenbankMultiple alignmentMaximum likelihood tree reconstructionVisualizing treesVisualizing trees and traitsMake fake traits:Visualize them on trees:</p>
  </div>
  <footer class="entry-footer"><span title='2011-02-21 12:31:00.001 -0600 -0600'>February 21, 2011</span>&nbsp;·&nbsp;1 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Phylogenetic analysis with the phangorn package: an example" href="http://localhost:1313/2011/02/phylogenetic-analysis-with-the-phangorn-package-an-example/"></a>
</article>

<article class="post-entry tag-entry"> 
  <header class="entry-header">
    <h2 class="entry-hint-parent">Troubling news for the teaching of evolution
    </h2>
  </header>
  <div class="entry-content">
    <p>[UPDATE: i remade the maps in green, hope that helps…]
A recent survey reported in Science (“Defeating Creationism in the Courtroom, but not in the Classroom”) found that biology teachers in high school do not often accept the basis of their discipline, as do teachers in other disciplines, and thus may not teach evolution appropriately. Read more here: New York Times.
I took a little time to play with the data provided online along with the Science article....</p>
  </div>
  <footer class="entry-footer"><span title='2011-02-09 08:20:00.009 -0600 -0600'>February 9, 2011</span>&nbsp;·&nbsp;3 min&nbsp;·&nbsp;Scott Chamberlain</footer>
  <a class="entry-link" aria-label="post link to Troubling news for the teaching of evolution" href="http://localhost:1313/2011/02/troubling-news-for-teaching-of/"></a>
</article>
<footer class="page-footer">
  <nav class="pagination">
    <a class="prev" href="http://localhost:1313/tags/r/page/14/">
      «&nbsp;Prev&nbsp;
    </a>
    <a class="next" href="http://localhost:1313/tags/r/page/16/">Next&nbsp;&nbsp;»
    </a>
  </nav>
</footer>
    </main>
    
<footer class="footer">
    <span>&copy; 2024 <a href="http://localhost:1313/">Recology</a></span>
    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
